{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 71,
   "id": "80fcb08a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "from Bio import SeqIO\n",
    "from collections import defaultdict\n",
    "from sklearn.datasets import load_iris\n",
    "from sklearn import tree\n",
    "from sklearn.preprocessing import OneHotEncoder\n",
    "import glob\n",
    "from itertools import combinations\n",
    "import subprocess\n",
    "import io\n",
    "import os\n",
    "import pandas as pd\n",
    "\n",
    "\n",
    "def read_vcf(path):\n",
    "    with open(path, 'r') as f:\n",
    "        lines = [l for l in f if not l.startswith('##')]\n",
    "    return pd.read_csv(\n",
    "        io.StringIO(''.join(lines)),\n",
    "        dtype={'#CHROM': str, 'POS': int, 'ID': str, 'REF': str, 'ALT': str,\n",
    "               'QUAL': str, 'FILTER': str, 'INFO': str},\n",
    "        sep='\\t'\n",
    "    ).rename(columns={'#CHROM': 'CHROM'})"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ba3b2645",
   "metadata": {},
   "source": [
    "https://github.com/egonozer/in_silico_pcr\n",
    "    \n",
    "F GTCGGTAAAACTCGTGCCAGC\n",
    "\n",
    "R CATAGTGGGGTATCTAATCCCAGTTTG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "id": "419ab3bf",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#Seq_id        Length(bp)     Circularity    Closely_related_species\r\n",
      "k141_193187    16623          yes            Salvelinus leucomaenis\r\n",
      "\r\n",
      "\r\n",
      "#Seq_id        Start  End    Length(bp) Direction  Type   Gene_name  Gene_prodcut                      Total_freq_occurred\r\n",
      "--------------------------------------------------------------------------------------------------------------------------\r\n",
      "k141_193187    <1     1362   1362       -          CDS    ND5        NADH dehydrogenase subunit 5      1              \r\n",
      "k141_193187    1363   1435   73         -          tRNA   trnL(uag)  tRNA-Leu                          2              \r\n",
      "k141_193187    1437   1505   69         -          tRNA   trnS(gcu)  tRNA-Ser                          2              \r\n",
      "k141_193187    1506   1574   69         -          tRNA   trnH(gug)  tRNA-His                          1              \r\n",
      "k141_193187    1570   2955   1386       -          CDS    ND4        NADH dehydrogenase subunit 4      1              \r\n",
      "k141_193187    2949   3245   297        -          CDS    ND4L       NADH dehydrogenase subunit 4L     1              \r\n",
      "k141_193187    3246   3315   70         -          tRNA   trnR(ucg)  tRNA-Arg                          1              \r\n",
      "k141_193187    3314   3664   351        -          CDS    ND3        NADH dehydrogenase subunit 3      1              \r\n",
      "k141_193187    3665   3734   70         -          tRNA   trnG(ucc)  tRNA-Gly                          1              \r\n",
      "k141_193187    3734   4519   786        -          CDS    COX3       cytochrome c oxidase subunit III  1              \r\n",
      "k141_193187    4519   5202   684        -          CDS    ATP6       ATP synthase F0 subunit 6         1              \r\n",
      "k141_193187    5193   5360   168        -          CDS    ATP8       ATP synthase F0 subunit 8         1              \r\n",
      "k141_193187    5362   5435   74         -          tRNA   trnK(uuu)  tRNA-Lys                          1              \r\n",
      "k141_193187    5428   6126   699        -          CDS    COX2       cytochrome c oxidase subunit II   1              \r\n",
      "k141_193187    6141   6214   74         -          tRNA   trnD(guc)  tRNA-Asp                          1              \r\n",
      "k141_193187    6219   6289   71         +          tRNA   trnS(uga)  tRNA-Ser                          2              \r\n",
      "k141_193187    6290   7840   1551       -          CDS    COX1       cytochrome c oxidase subunit I    1              \r\n",
      "k141_193187    7842   7912   71         +          tRNA   trnY(gua)  tRNA-Tyr                          1              \r\n",
      "k141_193187    7913   7979   67         +          tRNA   trnC(gca)  tRNA-Cys                          1              \r\n",
      "k141_193187    8016   8088   73         +          tRNA   trnN(guu)  tRNA-Asn                          1              \r\n",
      "k141_193187    8090   8158   69         +          tRNA   trnA(ugc)  tRNA-Ala                          1              \r\n",
      "k141_193187    8160   8231   72         -          tRNA   trnW(uca)  tRNA-Trp                          1              \r\n",
      "k141_193187    8231   9280   1050       -          CDS    ND2        NADH dehydrogenase subunit 2      1              \r\n",
      "k141_193187    9281   9349   69         -          tRNA   trnM(cau)  tRNA-Met                          1              \r\n",
      "k141_193187    9349   9419   71         +          tRNA   trnQ(uug)  tRNA-Gln                          1              \r\n",
      "k141_193187    9417   9488   72         -          tRNA   trnI(gau)  tRNA-Ile                          1              \r\n",
      "k141_193187    9495   10469  975        -          CDS    ND1        NADH dehydrogenase subunit 1      1              \r\n",
      "k141_193187    10470  10544  75         -          tRNA   trnL(uaa)  tRNA-Leu                          2              \r\n",
      "k141_193187    10545  12210  1666       -          rRNA   l-rRNA     16S ribosomal RNA                 1              \r\n",
      "k141_193187    12224  12295  72         -          tRNA   trnV(uac)  tRNA-Val                          1              \r\n",
      "k141_193187    12296  13242  947        -          rRNA   s-rRNA     12S ribosomal RNA                 1              \r\n",
      "k141_193187    13243  13310  68         -          tRNA   trnF(gaa)  tRNA-Phe                          1              \r\n",
      "k141_193187    14275  14344  70         +          tRNA   trnP(ugg)  tRNA-Pro                          1              \r\n",
      "k141_193187    14344  14415  72         -          tRNA   trnT(ugu)  tRNA-Thr                          1              \r\n",
      "k141_193187    14416  15556  1141       -          CDS    CYTB       cytochrome b                      1              \r\n",
      "k141_193187    15560  15628  69         +          tRNA   trnE(uuc)  tRNA-Glu                          1              \r\n",
      "k141_193187    15629  16150  522        +          CDS    ND6        NADH dehydrogenase subunit 6      1              \r\n",
      "\r\n",
      "\r\n",
      "Protein coding genes totally found: 13\r\n",
      "tRNA genes totally found:           22\r\n",
      "rRNA genes totally found:            2\r\n",
      "--------------------------------------\r\n",
      "Genes totally found:                37\r\n"
     ]
    }
   ],
   "source": [
    "!cat /storage/home/nsl5160/scratch/brook_trout/mitoassembly/29012_19/mt_annotation/29012_19.29012_19.megahit.mitogenome.fa.result/summary.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "id": "216a368a",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p ../gene_windows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "f6b644bd",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata = pd.read_csv(\"../wilmurt_meta.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "c9a59cea",
   "metadata": {},
   "outputs": [],
   "source": [
    "for index, seq_record in enumerate(SeqIO.parse(\"../ingroup_and_outgroup_circularized.msa\", \"fasta\")): \n",
    "    i=0\n",
    "    actual=0\n",
    "    with open(\"../msa_to_29012_19_coords.genemap\",\"a\") as f:\n",
    "        while i < len(seq_record.seq):\n",
    "            f.write(str(actual+1) + \"\\t\" + str(i+1) + \"\\n\")\n",
    "            if seq_record.seq[i] != \"-\":\n",
    "                actual=actual+1\n",
    "            i = i+1\n",
    "    if index == 0:\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f0deec2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "for seq_record in SeqIO.parse(\"../ingroup_and_outgroup_circularized.msa\", \"fasta\"):\n",
    "        if str(seq_record.id) in hatchery:\n",
    "            for i in list(range(len(seq_record.seq) - 450))[0::5]:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "7d24eae1",
   "metadata": {},
   "outputs": [],
   "source": [
    "!rm -rf ../gene_windows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "id": "eab45857",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p ../gene_windows"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "db691e02",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/storage/group/ibb3/default/.conda/envs/masters/lib/python3.10/site-packages/Bio/Seq.py:2096: BiopythonDeprecationWarning: myseq.ungap(gap) is deprecated; please use myseq.replace(gap, \"\") instead.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "strains = ['Assinica (Hatchery)', 'Domestic (Hatchery)',\n",
    "       'Horn (Hatchery)', 'Little Tupper (Hatchery)', \n",
    "           'Temiscamie (Hatchery)', 'Windfall (Hatchery)']\n",
    "for comb in combinations(strains, 5):\n",
    "    combo = comb[0][0]+comb[1][0]+comb[2][0]+comb[3][0]+comb[4][0]\n",
    "    hatchery = (metadata[\"Novogene_ID\"][metadata[\"Source\"].isin(list(comb))]\n",
    "            [metadata[\"Source\"].str.contains(\"Hatchery\")]\n",
    "            [~metadata[\"Novogene_ID\"].str.contains(\"TxD\")]).tolist()\n",
    "    for seq_record in SeqIO.parse(\"../ingroup_and_outgroup_circularized.msa\", \"fasta\"):\n",
    "        if str(seq_record.id) in hatchery:\n",
    "            for i in list(range(len(seq_record.seq) - 350))[0::5]:\n",
    "                with open(\"../gene_windows/\" + str(i) + \"_\" + combo + \".fasta\",\"a\") as f:\n",
    "                   f.write(\">\" + str(seq_record.id) + \"\\n\")\n",
    "                   f.write(str(seq_record.seq[i:i+350].ungap(\"-\")) + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "id": "ef72162e",
   "metadata": {},
   "outputs": [],
   "source": [
    "strains = ['Assinica (Hatchery)', 'Domestic (Hatchery)',\n",
    "       'Horn (Hatchery)', 'Little Tupper (Hatchery)', \n",
    "           'Temiscamie (Hatchery)', 'Windfall (Hatchery)']\n",
    "for comb in combinations(strains, 4):\n",
    "    combo = comb[0][0]+comb[1][0]+comb[2][0]+comb[3][0]\n",
    "    hatchery = (metadata[\"Novogene_ID\"][metadata[\"Source\"].isin(list(comb))]\n",
    "            [metadata[\"Source\"].str.contains(\"Hatchery\")]\n",
    "            [~metadata[\"Novogene_ID\"].str.contains(\"TxD\")]).tolist()\n",
    "    for seq_record in SeqIO.parse(\"../ingroup_and_outgroup_circularized.msa\", \"fasta\"):\n",
    "        if str(seq_record.id) in hatchery:\n",
    "            for i in list(range(len(seq_record.seq) - 350))[0::5]:\n",
    "                with open(\"../gene_windows/\" + str(i) + \"_\" + combo + \".fasta\",\"a\") as f:\n",
    "                   f.write(\">\" + str(seq_record.id) + \"\\n\")\n",
    "                   f.write(str(seq_record.seq[i:i+350].ungap(\"-\")) + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "id": "f7b219f2",
   "metadata": {},
   "outputs": [],
   "source": [
    "strains = ['Assinica (Hatchery)', 'Domestic (Hatchery)',\n",
    "       'Horn (Hatchery)', 'Little Tupper (Hatchery)', \n",
    "           'Temiscamie (Hatchery)', 'Windfall (Hatchery)']\n",
    "for comb in combinations(strains, 3):\n",
    "    combo = comb[0][0]+comb[1][0]+comb[2][0]\n",
    "    hatchery = (metadata[\"Novogene_ID\"][metadata[\"Source\"].isin(list(comb))]\n",
    "            [metadata[\"Source\"].str.contains(\"Hatchery\")]\n",
    "            [~metadata[\"Novogene_ID\"].str.contains(\"TxD\")]).tolist()\n",
    "    for seq_record in SeqIO.parse(\"../ingroup_and_outgroup_circularized.msa\", \"fasta\"):\n",
    "        if str(seq_record.id) in hatchery:\n",
    "            for i in list(range(len(seq_record.seq) - 350))[0::5]:\n",
    "                with open(\"../gene_windows/\" + str(i) + \"_\" + combo + \".fasta\",\"a\") as f:\n",
    "                   f.write(\">\" + str(seq_record.id) + \"\\n\")\n",
    "                   f.write(str(seq_record.seq[i:i+350].ungap(\"-\")) + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "id": "c945c745",
   "metadata": {},
   "outputs": [],
   "source": [
    "hatchery = metadata[\"Novogene_ID\"][metadata[\"Source\"].str.contains(\"Hatchery\")][~metadata[\"Novogene_ID\"].str.contains(\"TxD\")].tolist()\n",
    "for seq_record in SeqIO.parse(\"../ingroup_and_outgroup_circularized.msa\", \"fasta\"):\n",
    "    if str(seq_record.id) in hatchery:\n",
    "        for i in list(range(len(seq_record.seq) - 350))[0::5]:\n",
    "            with open(\"../gene_windows/\" + str(i) + \".fasta\",\"a\") as f:\n",
    "               f.write(\">\" + str(seq_record.id) + \"\\n\")\n",
    "               f.write(str(seq_record.seq[i:i+350].ungap(\"-\")) + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "dadd1719",
   "metadata": {},
   "outputs": [],
   "source": [
    "def split(a, n):\n",
    "    k, m = divmod(len(a), n)\n",
    "    return (a[i*k+min(i, m):(i+1)*k+min(i+1, m)] for i in range(n))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "ab2b446e-7a71-4822-9285-da9b7e14d9d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir ../primer_design"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "id": "a837b7f2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 7294164\n",
      "Submitted batch job 7294165\n",
      "Submitted batch job 7294166\n",
      "Submitted batch job 7294167\n",
      "Submitted batch job 7294168\n",
      "Submitted batch job 7294169\n",
      "Submitted batch job 7294170\n",
      "Submitted batch job 7294171\n",
      "Submitted batch job 7294172\n",
      "Submitted batch job 7294173\n",
      "Submitted batch job 7294174\n",
      "Submitted batch job 7294175\n",
      "Submitted batch job 7294176\n",
      "Submitted batch job 7294177\n",
      "Submitted batch job 7294178\n",
      "Submitted batch job 7294179\n",
      "Submitted batch job 7294180\n",
      "Submitted batch job 7294181\n",
      "Submitted batch job 7294182\n",
      "Submitted batch job 7294183\n",
      "Submitted batch job 7294184\n",
      "Submitted batch job 7294185\n",
      "Submitted batch job 7294186\n",
      "Submitted batch job 7294187\n",
      "Submitted batch job 7294188\n",
      "Submitted batch job 7294189\n",
      "Submitted batch job 7294190\n",
      "Submitted batch job 7294191\n",
      "Submitted batch job 7294192\n",
      "Submitted batch job 7294193\n",
      "Submitted batch job 7294194\n",
      "Submitted batch job 7294195\n",
      "Submitted batch job 7294196\n",
      "Submitted batch job 7294197\n",
      "Submitted batch job 7294198\n",
      "Submitted batch job 7294199\n",
      "Submitted batch job 7294200\n",
      "Submitted batch job 7294201\n",
      "Submitted batch job 7294202\n",
      "Submitted batch job 7294203\n",
      "Submitted batch job 7294204\n",
      "Submitted batch job 7294205\n",
      "Submitted batch job 7294206\n",
      "Submitted batch job 7294207\n",
      "Submitted batch job 7294208\n",
      "Submitted batch job 7294209\n",
      "Submitted batch job 7294210\n",
      "Submitted batch job 7294211\n",
      "Submitted batch job 7294212\n",
      "Submitted batch job 7294213\n",
      "Submitted batch job 7294214\n",
      "Submitted batch job 7294215\n",
      "Submitted batch job 7294216\n",
      "Submitted batch job 7294217\n",
      "Submitted batch job 7294218\n",
      "Submitted batch job 7294219\n",
      "Submitted batch job 7294220\n",
      "Submitted batch job 7294221\n",
      "Submitted batch job 7294222\n",
      "Submitted batch job 7294223\n",
      "Submitted batch job 7294224\n",
      "Submitted batch job 7294225\n",
      "Submitted batch job 7294226\n",
      "Submitted batch job 7294227\n",
      "Submitted batch job 7294228\n",
      "Submitted batch job 7294229\n",
      "Submitted batch job 7294230\n",
      "Submitted batch job 7294231\n",
      "Submitted batch job 7294232\n",
      "Submitted batch job 7294233\n",
      "Submitted batch job 7294234\n",
      "Submitted batch job 7294235\n",
      "Submitted batch job 7294236\n",
      "Submitted batch job 7294237\n",
      "Submitted batch job 7294238\n",
      "Submitted batch job 7294239\n",
      "Submitted batch job 7294240\n",
      "Submitted batch job 7294241\n",
      "Submitted batch job 7294242\n",
      "Submitted batch job 7294243\n",
      "Submitted batch job 7294244\n",
      "Submitted batch job 7294245\n",
      "Submitted batch job 7294246\n",
      "Submitted batch job 7294247\n",
      "Submitted batch job 7294248\n",
      "Submitted batch job 7294249\n",
      "Submitted batch job 7294250\n",
      "Submitted batch job 7294251\n",
      "Submitted batch job 7294252\n",
      "Submitted batch job 7294253\n",
      "Submitted batch job 7294254\n",
      "Submitted batch job 7294255\n",
      "Submitted batch job 7294256\n",
      "Submitted batch job 7294257\n",
      "Submitted batch job 7294258\n",
      "Submitted batch job 7294259\n",
      "Submitted batch job 7294260\n",
      "Submitted batch job 7294261\n",
      "Submitted batch job 7294262\n",
      "Submitted batch job 7294263\n"
     ]
    }
   ],
   "source": [
    "for n_strains in [3]:\n",
    "    for fragsize in [200]:\n",
    "        for i in list(split(range(0,16663-fragsize), 100)):\n",
    "            #for i in list(split(range(0,300-fragsize), 10)):\n",
    "            task = \"primer_chunk\"\n",
    "            mem = \"8\"\n",
    "            cpus = \"1\"\n",
    "            start = list(i)[0]\n",
    "            end = list(i)[-1]\n",
    "            subprocess.run([\"sbatch --mem=\" + mem + \"g --ntasks=\" + \n",
    "                 cpus + \" ../\" + task + \".sh \" + \n",
    "                            str(start) + \" \" + str(end) + \" \" + str(fragsize) + \" \" + str(n_strains)], shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "id": "8a1e83fd-aa28-401f-9501-f6ea37da8628",
   "metadata": {},
   "outputs": [],
   "source": [
    "strains = ['Assinica (Hatchery)', 'Domestic (Hatchery)',\n",
    "       'Horn (Hatchery)', 'Little Tupper (Hatchery)', \n",
    "           'Temiscamie (Hatchery)', 'Windfall (Hatchery)']"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6114202-98c4-4d3a-9e6d-e6fd8fb8efb5",
   "metadata": {},
   "source": [
    "\n",
    "for comb in combinations(strains, 5):\n",
    "    combo = comb[0][0]+comb[1][0]+comb[2][0]+comb[3][0]+comb[4][0]\n",
    "    hatchery = (metadata[\"Novogene_ID\"][metadata[\"Source\"].isin(list(comb))]\n",
    "            [metadata[\"Source\"].str.contains(\"Hatchery\")]\n",
    "            [~metadata[\"Novogene_ID\"].str.contains(\"TxD\")]).tolist()"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e401b8df-445e-4a33-a929-433eac02fe27",
   "metadata": {},
   "source": [
    "for comb in combinations(strains, 5):\n",
    "    combo = comb[0][0]+comb[1][0]+comb[2][0]+comb[3][0]+comb[4][0]\n",
    "    hatchery = (metadata[\"Novogene_ID\"][metadata[\"Source\"].isin(list(comb))]\n",
    "            [metadata[\"Source\"].str.contains(\"Hatchery\")]\n",
    "            [~metadata[\"Novogene_ID\"].str.contains(\"TxD\")]).tolist()\n",
    "    for seq_record in SeqIO.parse(\"../ingroup_and_outgroup_circularized.msa\", \"fasta\"):\n",
    "        if str(seq_record.id) in hatchery:\n",
    "            start = 0\n",
    "            length = 200\n",
    "            try:\n",
    "                with open('/storage/home/nsl5160/scratch/primer_design/tmp.fasta', 'x') as f:\n",
    "                    pass\n",
    "            except FileExistsError:\n",
    "                with open(\"/storage/home/nsl5160/scratch/primer_design/tmp.fasta\",\"a\") as f:\n",
    "                   f.write(\">\" + str(seq_record.id) + \"\\n\")\n",
    "                   f.write(str(seq_record.seq[start:start+length].ungap(\"-\")) + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "id": "80688157",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata[metadata[\"Source\"].isin(strains)].to_csv(\"../hatchery_only.csv\", sep = \",\", index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "id": "b97941b0",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"../hatchery_only.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "id": "3ba70a86",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 4s, sys: 781 ms, total: 1min 4s\n",
      "Wall time: 1min 8s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "seeds = []\n",
    "for window in [i.split(\"/\")[3].split(\".\")[0] for i in glob.glob(\"../primer_design/window_200/HTW_*.derep.uc\")]:\n",
    "    tmp = pd.read_csv(\"../primer_design/window_200/\" + window + \".derep.uc\", sep = \"\\t\", header = None)\n",
    "    tmp = tmp[tmp[9] != \"*\"]\n",
    "    tmp[\"hit\"] = (tmp[8].str.replace(\"Assinica_\", \"A\").str.replace(\"Horn_\", \"H\").str.replace(\"Dom_\", \"D\")\n",
    "     .str.replace(\"LTup_\", \"L\").str.replace(\"Temi_\", \"T\").str.replace(\"Wind_\", \"W\").str.replace(\"Ass_\", \"A\"))\n",
    "    tmp[\"strain_id_hit\"] = (tmp[8].str.replace(\"Assinica_\", \"A\").str.replace(\"Horn_\", \"H\").str.replace(\"Dom_\", \"D\")\n",
    "     .str.replace(\"LTup_\", \"L\").str.replace(\"Temi_\", \"T\").str.replace(\"Wind_\", \"W\").str.replace(\"Ass_\", \"A\").str.split(\"\", expand =True))[1]\n",
    "    tmp[\"seed\"] = (tmp[9].str.replace(\"Assinica_\", \"A\").str.replace(\"Horn_\", \"H\").str.replace(\"Dom_\", \"D\")\n",
    "     .str.replace(\"LTup_\", \"L\").str.replace(\"Temi_\", \"T\").str.replace(\"Wind_\", \"W\").str.replace(\"Ass_\", \"A\"))\n",
    "    tmp[\"strain_id_seed\"] = (tmp[9].str.replace(\"Assinica_\", \"A\").str.replace(\"Horn_\", \"H\").str.replace(\"Dom_\", \"D\")\n",
    "     .str.replace(\"LTup_\", \"L\").str.replace(\"Temi_\", \"T\").str.replace(\"Wind_\", \"W\").str.replace(\"Ass_\", \"A\").str.split(\"\", expand =True))[1]\n",
    "    for i in tmp[9].unique():\n",
    "        tmp1 = tmp[tmp[9] == i].reset_index(drop=True)\n",
    "        tmplist = [tmp1[\"seed\"][0]] + tmp1[\"hit\"].tolist()\n",
    "        tmpstrains = list(set([tmp1[\"strain_id_seed\"][0]] + tmp1[\"strain_id_hit\"].tolist()))\n",
    "        seeds = seeds + [[window, i, tmp1[\"seed\"][0], \"|\".join(tmplist), tmp1[\"strain_id_seed\"][0], \"|\".join(tmpstrains)]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "08030fe9",
   "metadata": {},
   "outputs": [],
   "source": [
    "seeds = pd.DataFrame(seeds)\n",
    "seeds[\"start\"] = seeds[0].str.split(\"_\", expand = True)[1].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "id": "ca1aa8e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "mask = (seeds[5].str.len() == 1)\n",
    "tmp = seeds.loc[mask].sort_values(by=3, key=lambda x: x.str.len(), ascending = False).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "id": "1abefbaf",
   "metadata": {},
   "outputs": [],
   "source": [
    "import itertools\n",
    "merged = list(set(list(itertools.chain(*[i.split(\"|\") for i in seeds[3].tolist()]))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "c142bbe9",
   "metadata": {},
   "outputs": [],
   "source": [
    "H = [i for i in merged if \"H\" in i]\n",
    "W = [i for i in merged if \"W\" in i]\n",
    "T = [i for i in merged if \"T\" in i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "69b75a0c",
   "metadata": {},
   "outputs": [],
   "source": [
    "informative = pd.DataFrame()\n",
    "for i in tmp[3].unique():\n",
    "    test = tmp[tmp[0].isin(tmp[0][tmp[3] == i].tolist())]\n",
    "    for k in test[0][test.duplicated(0, keep=False) == True].tolist():\n",
    "        if len(test[test[0] == k]) > 0:\n",
    "            informative = pd.concat([informative, pd.DataFrame(data = [i, k, len(test[test[0] == k]), \",\".join(list(set(test[3][test[0] == k].tolist())))]).T]).reset_index(drop=True)\n",
    "    if len(test[0][test.duplicated(0, keep=False) == True].tolist()) < 1:\n",
    "        test1 = test.copy().reset_index(drop=True)\n",
    "        for k in range(len(test1)):\n",
    "            informative = pd.concat([informative, pd.DataFrame(data = [i, test1[0][k], 1, i]).T]).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "5fb3f753",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('T13|T14|T15|T1|T10|T11|T12|T2|T3|T4|T6|T7|T8|T9,H1|H11|H12|H14|H3|H4|H5|H7|H8|H9', 'H10|H13|H2|H6') 1.0 0.0 0.93\n"
     ]
    }
   ],
   "source": [
    "for combo in combinations(informative[3].unique(), 2):\n",
    "    multiplex = list(set(\"|\".join(list(combo)).replace(\",\", \"|\").split(\"|\")))\n",
    "    h_coverage = round(len([i for i in multiplex if i in H]) / len(H), 2)\n",
    "    w_coverage = round(len([i for i in multiplex if i in W]) / len(W), 2)\n",
    "    t_coverage = round(len([i for i in multiplex if i in T]) / len(T), 2)\n",
    "    if (h_coverage == 1 or w_coverage == 1 or t_coverage == 1):\n",
    "        print(combo, h_coverage, w_coverage, t_coverage)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3559d21b",
   "metadata": {},
   "source": [
    "#informative = informative.drop_duplicates(1).reset_index(drop=True)\n",
    "informative = informative.drop_duplicates(3).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "f3ab3413",
   "metadata": {},
   "outputs": [],
   "source": [
    "hatchery = metadata[\"Novogene_ID\"][metadata[\"Source\"].str.contains(\"Hatchery\")][~metadata[\"Novogene_ID\"].str.contains(\"TxD\")].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "e4380708",
   "metadata": {},
   "outputs": [],
   "source": [
    "testset = informative[informative[3].isin([informative[3].drop_duplicates().tolist()[1], informative[3].drop_duplicates().tolist()[3]])].drop_duplicates(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "1ee7179d",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "testset = testset.reset_index(drop=True)\n",
    "testset[\"start\"] = testset[1].str.split(\"_\", expand = True)[1].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "id": "28e33e87",
   "metadata": {},
   "outputs": [],
   "source": [
    "testset = testset.sort_values(\"start\").reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "c5f4ba1a-4049-4421-828f-b9e117b33cd1",
   "metadata": {},
   "outputs": [],
   "source": [
    "primer_candidates = pd.DataFrame()\n",
    "for primerlengthF in [15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25]:\n",
    "    for primerlengthR in [15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25]:\n",
    "        for i in testset[\"start\"].tolist():\n",
    "            varseq = []\n",
    "            invarseqF = []\n",
    "            invarseqR = []\n",
    "            for seq_record in SeqIO.parse(\"../ingroup_and_outgroup_circularized.msa\", \"fasta\"):\n",
    "                if str(seq_record.id) in hatchery:\n",
    "                    varseq = varseq + [str(seq_record.seq)[i:i+200].replace(\"-\", \"\")]\n",
    "                    invarseqF = invarseqF + [str(seq_record.seq)[i-primerlengthF:i].replace(\"-\", \"\")]\n",
    "                    invarseqR = invarseqR + [str(seq_record.seq)[i+200:i+200+primerlengthR].replace(\"-\", \"\")]\n",
    "            if len(set(invarseqF)) == 1 and len(set(invarseqR)) == 1:\n",
    "                tmp = primer_summary(invarseqF[0].upper(), invarseqR[0].upper())\n",
    "                tmp = pd.DataFrame(data = [testset[3][testset[\"start\"] == i].reset_index(drop=True)[0], \n",
    "                                           testset[\"start\"][testset[\"start\"] == i].reset_index(drop=True)[0], \n",
    "                                           primerlengthF, primerlengthR, \n",
    "                                           invarseqF[0].upper(), invarseqR[0].upper()] + \n",
    "                                   [z for y in tmp for z in y]).T\n",
    "                tmp.columns = [\"Target Group\", \"Window Start Pos\", \"Primer 1 Length\", \"Primer 2 Length\", \"Primer 1 (+ strand, 5'>3')\", \"Primer 2 (+ strand, 5'>3')\", \n",
    "                               \"P1 Melting\", \"P1 PolyX\", \n",
    "                               \"P1 GC\", \"P2 Melting\", \"P2 PolyX\", \n",
    "                               \"P2 GC\", \"P1_P1_3prime_complementarity\", \n",
    "                               \"P2_P2_3prime_complementarity\", \"P1_P2_3prime_complementarity\", \n",
    "                               \"P1 Clamp\", \"P2 Clamp\"]\n",
    "                primer_candidates = pd.concat([primer_candidates, tmp]).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "47545eba",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/451018082.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n"
     ]
    }
   ],
   "source": [
    "bestcandidates = (primer_candidates[primer_candidates[\"P1 GC\"] > 0.4][primer_candidates[\"P1 GC\"] < 0.6]\n",
    " [primer_candidates[\"P2 GC\"] > 0.4][primer_candidates[\"P2 GC\"] < 0.6]\n",
    " [primer_candidates[\"P1 PolyX\"] == 0][primer_candidates[\"P2 PolyX\"] == 0]\n",
    " [primer_candidates[\"P1_P1_3prime_complementarity\"] == False]\n",
    " [primer_candidates[\"P2_P2_3prime_complementarity\"] == False]\n",
    " [primer_candidates[\"P1_P2_3prime_complementarity\"] == False]\n",
    " [primer_candidates[\"P1 Clamp\"] == True][primer_candidates[\"P2 Clamp\"] == True]\n",
    " [primer_candidates[\"P1 Melting\"] < 65][primer_candidates[\"P1 Melting\"] > 50]\n",
    " [primer_candidates[\"P2 Melting\"] < 65][primer_candidates[\"P2 Melting\"] > 50])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "id": "e20ab37b",
   "metadata": {},
   "outputs": [],
   "source": [
    "bestcandidates = bestcandidates.loc[abs(bestcandidates[\"P1 Melting\"] - bestcandidates[\"P2 Melting\"]) < 3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "id": "3577342b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/2480618833.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  bestcandidates[\"Primer F\"] = bestcandidates[\"Primer 1 (+ strand, 5\\'>3\\')\"]\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/2480618833.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  bestcandidates[\"Primer R\"] = bestcandidates[\"Primer 2 (+ strand, 5\\'>3\\')\"].apply(lambda x: x[::-1]).apply(lambda x: x.translate(str.maketrans(\"ATGC\", \"TACG\")))\n"
     ]
    }
   ],
   "source": [
    "bestcandidates[\"Primer F\"] = bestcandidates[\"Primer 1 (+ strand, 5\\'>3\\')\"]\n",
    "bestcandidates[\"Primer R\"] = bestcandidates[\"Primer 2 (+ strand, 5\\'>3\\')\"].apply(lambda x: x[::-1]).apply(lambda x: x.translate(str.maketrans(\"ATGC\", \"TACG\")))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "447572b2",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/2760153910.py:1: UserWarning: Boolean Series key will be reindexed to match DataFrame index.\n",
      "  bestcandidates = bestcandidates[bestcandidates[\"Primer 1 Length\"] > 18][bestcandidates[\"Primer 2 Length\"] > 18].reset_index(drop=True)\n"
     ]
    }
   ],
   "source": [
    "bestcandidates = bestcandidates[bestcandidates[\"Primer 1 Length\"] > 18][bestcandidates[\"Primer 2 Length\"] > 18].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "id": "068ae430",
   "metadata": {},
   "outputs": [],
   "source": [
    "bestcandidates.to_csv(\"../primer_candidates.tsv\", sep = \"\\t\", index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "5e968ecc",
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"../primer1.fasta\",\"a\") as f:\n",
    "    for i in range(len(bestcandidates)):\n",
    "       f.write(\">\" + str(i) + \"\\n\")\n",
    "       f.write(bestcandidates[\"Primer 1 (+ strand, 5\\'>3\\')\"][i] + \"\\n\")\n",
    "with open(\"../primer2.fasta\",\"a\") as f:\n",
    "    for i in range(len(bestcandidates)):\n",
    "       f.write(\">\" + str(i) + \"\\n\")\n",
    "       f.write(bestcandidates[\"Primer 2 (+ strand, 5\\'>3\\')\"][i] + \"\\n\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d7e02a62-46be-4d13-97b6-463441221211",
   "metadata": {},
   "source": [
    "The coordinate start positions need to be updated to that of a RefSeq sequence so that we can run primerBLAST on the regions that appear to be best in our first pass. We'll start this by aligning one of our sequences against NC_000860.1 using MAFFT. This will give us an idea of how the start and stop positions need to be adjusted:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f949dccc-dabf-4259-8720-3dc03d4dd931",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "cat ../NC_000860.1.fasta > ../offset.fa\n",
    "awk '/^>/{if(N)exit;++N;} {print;}' ../ingroup_and_outgroup_circularized.msa | sed \"s:-::g\" \\\n",
    "    >> ../offset.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d034eab1-aef8-47fd-8d9b-e5c0d74dbac3",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "source ~/.bashrc\n",
    "conda activate orbicella\n",
    "mafft ../offset.fa > ../offset.msa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "1ccb99d4-1756-4bf2-b91b-7dd6f7f365b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "for seq_record in SeqIO.parse(\"../offset.msa\", \"fasta\"):\n",
    "    if \"NC_0008\" not in str(seq_record.id):\n",
    "        i = 1\n",
    "        pos = \"-\"\n",
    "        while pos == \"-\":\n",
    "            pos = str(seq_record.seq[i])\n",
    "            i = i + 1\n",
    "    else:\n",
    "        refseq = seq_record"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "9888e9c9-f096-4903-a176-cd621872e065",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1059"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "i-1"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c4d975d7-3429-49d5-8204-06e0c2d205ab",
   "metadata": {},
   "source": [
    "It looks like the offset should be 1059 bases, so when we run primer blast, we should give it windows to design primers within that are offset by 1060 bases from our initial set:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "id": "dcec41fa-cf3e-461e-b59f-c53cb8c3f39c",
   "metadata": {},
   "outputs": [],
   "source": [
    "candidates = pd.read_csv(\"../primer_candidates.tsv\", sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "id": "d25ec7d6-aa1f-4f1e-a83e-6e9887655d83",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "candidates[\"Primer 1 Start Position (NC_000860.1), 1-based\"] = (candidates[\"Window Start Pos\"] - \n",
    "                                                                candidates[\"Primer 1 Length\"] + 1059)\n",
    "candidates[\"Primer 1 End Position (NC_000860.1), 1-based\"] = (candidates[\"Window Start Pos\"] + 1059)\n",
    "candidates[\"Primer 2 Start Position (NC_000860.1), 1-based\"] = (candidates[\"Window Start Pos\"] + \n",
    "                                                                200 + 1059)\n",
    "candidates[\"Primer 2 End Position (NC_000860.1), 1-based\"] = (candidates[\"Window Start Pos\"] + \n",
    "                                                                200 + \n",
    "                                                                candidates[\"Primer 2 Length\"] + 1059)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "5a1b5050-341e-484f-afc6-eca55489beaa",
   "metadata": {},
   "outputs": [],
   "source": [
    "for seq_record in SeqIO.parse(\"../ingroup_and_outgroup_circularized.msa\", \"fasta\"):\n",
    "    if str(seq_record.id) == \"29012_19\":\n",
    "        seq = seq_record"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "5aa6c627-da38-418a-8b92-f5d90fdf54c5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SeqRecord(seq=Seq('ccagcaaaccagagcaaag'), id='NC_000860.1', name='NC_000860.1', description='NC_000860.1 Salvelinus fontinalis mitochondrion, complete genome', dbxrefs=[])"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "refseq[2280:2299]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "3f825da7-04eb-4082-aa3b-455055d39a51",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SeqRecord(seq=Seq('accttgtcccaaagaaacc'), id='NC_000860.1', name='NC_000860.1', description='NC_000860.1 Salvelinus fontinalis mitochondrion, complete genome', dbxrefs=[])"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "refseq[2499:2518]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "ed1dc0b6-b916-4572-9a1e-e00299dbd243",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "candidates = candidates.drop_duplicates(\"Window Start Pos\").reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "id": "5b0992e2-1fe2-418c-bfc7-e4555f887edb",
   "metadata": {},
   "outputs": [],
   "source": [
    "i = 0\n",
    "windows = [testset[\"start\"][0]]\n",
    "for i in range(len(testset)):\n",
    "    try:\n",
    "        if (testset[\"start\"][i] - testset[\"start\"][i-1]) > 1:\n",
    "            windows = windows + [testset[\"start\"][i-1], testset[\"start\"][i]]\n",
    "    except:\n",
    "        pass\n",
    "windows = windows + [testset[\"start\"][i]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "5999a53b-091d-4dd7-b224-1a57e3932eb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "windows = [i+1059 for i in windows]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "86c6f0a4-68b5-4ba2-b33b-23752d69f7e5",
   "metadata": {},
   "outputs": [],
   "source": [
    "window1 = 2283-25, 2375, 2283+200, 2375+200+25\n",
    "window2 = 10459-25, 10658, 10459+200, 10658+200+25"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7ad3fc37-2b85-4d8c-a8db-a6b44919bdc6",
   "metadata": {},
   "source": [
    "We ran searches externally using Primer-BLAST using the accession NC_000860.1 using the above primer windows. Max PCR product size allowed was 250bp and returned max 100 primers per search. Search database was core_nt and search was performed on eukaryotes, archaea, and bacteria. All other parameters were left as default. I've uploaded the primer results here as set1.csv for target group 'T13|T14|T15|T1|T10|T11|T12|T2|T3|T4|T6|T7|T8|T9,H1|H11|H12|H14|H3|H4|H5|H7|H8|H9' and set2.csv for target group 'H10|H13|H2|H6'. Set1 targets the l-rRNA (16s) and set2 targets ND3 (although many set2 F primers sit solidly in tRNA-Gly). I'm going to re-run some digests here to make sure they actually do what I'm hoping and that they highlight the differences between strains:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "id": "2753acd3-31dc-4438-aa74-7adfd6b874e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "set1 = pd.read_csv(\"../set1.csv\")\n",
    "set2 = pd.read_csv(\"../set2.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "4b0da517-7137-4493-a58a-4bcfba81caf2",
   "metadata": {},
   "outputs": [],
   "source": [
    "for comb in [('Horn (Hatchery)', 'Temiscamie (Hatchery)', 'Windfall (Hatchery)')]:\n",
    "    combo = \"\".join([word[0] for word in list(comb)])\n",
    "    hatchery = (metadata[\"Novogene_ID\"][metadata[\"Source\"].isin(list(comb))]).tolist()\n",
    "    for seq_record in SeqIO.parse(\"../ingroup_and_outgroup_circularized.fa\", \"fasta\"):\n",
    "        if str(seq_record.id) in hatchery:\n",
    "            with open(\"hatchery.fasta\", \"a\") as f:\n",
    "               f.write(\">\" + str(seq_record.id) + \"\\n\")\n",
    "               f.write(str(seq_record.seq) + \"\\n\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "id": "b3cf9011-3fc1-4997-b350-a080642fe8af",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8668 nt in 44 seqs, min 197, max 197, avg 197\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8712 nt in 44 seqs, min 198, max 198, avg 198\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8536 nt in 44 seqs, min 194, max 194, avg 194\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7304 nt in 44 seqs, min 166, max 166, avg 166\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8272 nt in 44 seqs, min 188, max 188, avg 188\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8712 nt in 44 seqs, min 198, max 198, avg 198\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8624 nt in 44 seqs, min 196, max 196, avg 196\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7700 nt in 44 seqs, min 175, max 175, avg 175\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8624 nt in 44 seqs, min 196, max 196, avg 196\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8360 nt in 44 seqs, min 190, max 190, avg 190\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7612 nt in 44 seqs, min 173, max 173, avg 173\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8228 nt in 44 seqs, min 187, max 187, avg 187\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8536 nt in 44 seqs, min 194, max 194, avg 194\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8712 nt in 44 seqs, min 198, max 198, avg 198\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8712 nt in 44 seqs, min 198, max 198, avg 198\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8712 nt in 44 seqs, min 198, max 198, avg 198\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8668 nt in 44 seqs, min 197, max 197, avg 197\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8492 nt in 44 seqs, min 193, max 193, avg 193\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7480 nt in 44 seqs, min 170, max 170, avg 170\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8536 nt in 44 seqs, min 194, max 194, avg 194\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7612 nt in 44 seqs, min 173, max 173, avg 173\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8800 nt in 44 seqs, min 200, max 200, avg 200\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8316 nt in 44 seqs, min 189, max 189, avg 189\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8580 nt in 44 seqs, min 195, max 195, avg 195\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8800 nt in 44 seqs, min 200, max 200, avg 200\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8800 nt in 44 seqs, min 200, max 200, avg 200\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "5100 nt in 30 seqs, min 170, max 170, avg 170\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 15.0, median 15, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7480 nt in 44 seqs, min 170, max 170, avg 170\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8580 nt in 44 seqs, min 195, max 195, avg 195\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8712 nt in 44 seqs, min 198, max 198, avg 198\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8756 nt in 44 seqs, min 199, max 199, avg 199\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8756 nt in 44 seqs, min 199, max 199, avg 199\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8712 nt in 44 seqs, min 198, max 198, avg 198\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7700 nt in 44 seqs, min 175, max 175, avg 175\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8272 nt in 44 seqs, min 188, max 188, avg 188\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8316 nt in 44 seqs, min 189, max 189, avg 189\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8404 nt in 44 seqs, min 191, max 191, avg 191\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8492 nt in 44 seqs, min 193, max 193, avg 193\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7392 nt in 44 seqs, min 168, max 168, avg 168\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8756 nt in 44 seqs, min 199, max 199, avg 199\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8140 nt in 44 seqs, min 185, max 185, avg 185\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8800 nt in 44 seqs, min 200, max 200, avg 200\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8800 nt in 44 seqs, min 200, max 200, avg 200\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7700 nt in 44 seqs, min 175, max 175, avg 175\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8800 nt in 44 seqs, min 200, max 200, avg 200\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8800 nt in 44 seqs, min 200, max 200, avg 200\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7920 nt in 44 seqs, min 180, max 180, avg 180\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8184 nt in 44 seqs, min 186, max 186, avg 186\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8800 nt in 44 seqs, min 200, max 200, avg 200\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8624 nt in 44 seqs, min 196, max 196, avg 196\n",
      "Sorting 100%\n",
      "3 unique sequences, avg cluster 14.7, median 14, max 20\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=0) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "Making a distance matrix from msa.. \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 3\n",
      "done.\n",
      "\n",
      "Progressive alignment 2/2... \n",
      "STEP     2 / 2 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for i in range(len(set1)):\n",
    "    subprocess.run([\"source ~/.bashrc ; conda activate orbicella ; \" + \n",
    "                \"perl ../in_silico_PCR.pl \" + \n",
    "                \"-a \" + set1[\"Forward primer Sequence (5\\'->3\\')\"][i] + \n",
    "                \" -b \" + set1[\"Reverse primer Sequence (5\\'->3\\')\"][i] + \n",
    "                \" -s hatchery.fasta &> namemap.txt 2> amp.fa ; \" + \n",
    "                \"cut -f1-2 namemap.txt | tail -n+2 > rename.txt ; \" + \n",
    "                \"seqkit replace --quiet -p '(.+)$' -r '{kv}${2}' -k rename.txt amp.fa \" + \n",
    "                \"> tmp.fa ; \" + \"vsearch --derep_fulllength tmp.fa\" +\n",
    "                \" --output set1_\" + str(set1[\"Primer pair #\"][i]) + \"_derep.fasta\" +\n",
    "                \" --uc set1_\" + str(set1[\"Primer pair #\"][i]) + \".derep.uc\" +\n",
    "                \" --sizeout --threads 1 --fasta_width 0 ; mafft set1_\" + \n",
    "                str(set1[\"Primer pair #\"][i]) + \"_derep.fasta \" + \n",
    "                \"> set1_\" + str(set1[\"Primer pair #\"][i]) + \"_derep.msa\"], shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "id": "2cb83480-dee1-49e5-badb-d4d618c2cdf6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7744 nt in 44 seqs, min 176, max 176, avg 176\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7964 nt in 44 seqs, min 181, max 181, avg 181\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8184 nt in 44 seqs, min 186, max 186, avg 186\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8140 nt in 44 seqs, min 185, max 185, avg 185\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7920 nt in 44 seqs, min 180, max 180, avg 180\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7920 nt in 44 seqs, min 180, max 180, avg 180\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8184 nt in 44 seqs, min 186, max 186, avg 186\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8228 nt in 44 seqs, min 187, max 187, avg 187\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8096 nt in 44 seqs, min 184, max 184, avg 184\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7788 nt in 44 seqs, min 177, max 177, avg 177\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7744 nt in 44 seqs, min 176, max 176, avg 176\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7832 nt in 44 seqs, min 178, max 178, avg 178\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7876 nt in 44 seqs, min 179, max 179, avg 179\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7568 nt in 44 seqs, min 172, max 172, avg 172\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7700 nt in 44 seqs, min 175, max 175, avg 175\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8008 nt in 44 seqs, min 182, max 182, avg 182\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8228 nt in 44 seqs, min 187, max 187, avg 187\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8008 nt in 44 seqs, min 182, max 182, avg 182\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7876 nt in 44 seqs, min 179, max 179, avg 179\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7832 nt in 44 seqs, min 178, max 178, avg 178\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7832 nt in 44 seqs, min 178, max 178, avg 178\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7832 nt in 44 seqs, min 178, max 178, avg 178\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7524 nt in 44 seqs, min 171, max 171, avg 171\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8316 nt in 44 seqs, min 189, max 189, avg 189\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7832 nt in 44 seqs, min 178, max 178, avg 178\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7700 nt in 44 seqs, min 175, max 175, avg 175\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8580 nt in 44 seqs, min 195, max 195, avg 195\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7788 nt in 44 seqs, min 177, max 177, avg 177\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7744 nt in 44 seqs, min 176, max 176, avg 176\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8140 nt in 44 seqs, min 185, max 185, avg 185\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7700 nt in 44 seqs, min 175, max 175, avg 175\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7920 nt in 44 seqs, min 180, max 180, avg 180\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7920 nt in 44 seqs, min 180, max 180, avg 180\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8360 nt in 44 seqs, min 190, max 190, avg 190\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8096 nt in 44 seqs, min 184, max 184, avg 184\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8228 nt in 44 seqs, min 187, max 187, avg 187\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7700 nt in 44 seqs, min 175, max 175, avg 175\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7744 nt in 44 seqs, min 176, max 176, avg 176\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8360 nt in 44 seqs, min 190, max 190, avg 190\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7920 nt in 44 seqs, min 180, max 180, avg 180\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8756 nt in 44 seqs, min 199, max 199, avg 199\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8404 nt in 44 seqs, min 191, max 191, avg 191\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8624 nt in 44 seqs, min 196, max 196, avg 196\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8712 nt in 44 seqs, min 198, max 198, avg 198\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8800 nt in 44 seqs, min 200, max 200, avg 200\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "8624 nt in 44 seqs, min 196, max 196, avg 196\n",
      "Sorting 100%\n",
      "2 unique sequences, avg cluster 22.0, median 22, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "stacksize: -1 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "\n",
      "\n",
      "\n",
      "Making a distance matrix ..\n",
      "    1 / 2\n",
      "done.\n",
      "\n",
      "Constructing a UPGMA tree (efffree=1) ... \n",
      "    0 / 2\n",
      "done.\n",
      "\n",
      "Progressive alignment 1/1... \n",
      "STEP     1 / 1 \n",
      "done.\n",
      "\n",
      "disttbfast (nuc) Version 7.525\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-1 (Very fast but very rough)\n",
      " Progressive method (rough guide tree was used.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "vsearch v2.29.1_linux_x86_64, 753.3GB RAM, 128 cores\n",
      "https://github.com/torognes/vsearch\n",
      "\n",
      "Dereplicating file tmp.fa 100%\n",
      "7080 nt in 40 seqs, min 177, max 177, avg 177\n",
      "Sorting 100%\n",
      "1 unique sequences, avg cluster 40.0, median 40, max 40\n",
      "Writing FASTA output file 100%\n",
      "Writing uc file, first part 100%\n",
      "Writing uc file, second part 100%\n",
      "nthread = 0\n",
      "nthreadpair = 0\n",
      "nthreadtb = 0\n",
      "ppenalty_ex = 0\n",
      "Warning: Only 1 sequence found.\n",
      "\n",
      "Strategy:\n",
      " FFT-NS-2 (Fast but rough)\n",
      " Progressive method (guide trees were built 2 times.)\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "for i in range(len(set2)):\n",
    "    subprocess.run([\"source ~/.bashrc ; conda activate orbicella ; \" + \n",
    "                \"perl ../in_silico_PCR.pl \" + \n",
    "                \"-a \" + set2[\"Forward primer Sequence (5\\'->3\\')\"][i] + \n",
    "                \" -b \" + set2[\"Reverse primer Sequence (5\\'->3\\')\"][i] + \n",
    "                \" -s hatchery.fasta &> namemap.txt 2> amp.fa ; \" + \n",
    "                \"cut -f1-2 namemap.txt | tail -n+2 > rename.txt ; \" + \n",
    "                \"seqkit replace --quiet -p '(.+)$' -r '{kv}${2}' -k rename.txt amp.fa \" + \n",
    "                \"> tmp.fa ; \" + \"vsearch --derep_fulllength tmp.fa\" +\n",
    "                \" --output set2_\" + str(set2[\"Primer pair #\"][i]) + \"_derep.fasta\" +\n",
    "                \" --uc set2_\" + str(set2[\"Primer pair #\"][i]) + \".derep.uc\" +\n",
    "                \" --sizeout --threads 1 --fasta_width 0 ; mafft set2_\" + \n",
    "                str(set2[\"Primer pair #\"][i]) + \"_derep.fasta \" + \n",
    "                \"> set2_\" + str(set2[\"Primer pair #\"][i]) + \"_derep.msa\"], shell = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "id": "c3465fd5-d695-49d2-87ce-1c44d9d4a884",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Horn_10;size=20\n",
      "tatagggccaacccgtctctgtggcaaaagagtgggacgagccccgagtagaggtgataa\n",
      "acctatcgagcctagttatagctggttgcttaggaaatgaatagaagttcagccccctgg\n",
      "ctttcttaagaccccaaggtaaaactaaccttgtcccaaagaaaccaagagagttaatca\n",
      "aaggaggtacagctcct\n",
      ">T13;size=14\n",
      "tatagggccaacccgtctctgtagcaaaagagtgggacgagccccgagtagaggtgataa\n",
      "acctatcgagcctagttatagctggttgcttaggaaatgaatagaagttcagccccctgg\n",
      "ctttcttaagaccccaaggtaaaactaaccttgtcccaaagaaaccaagagagttaatca\n",
      "aaggaggtacagctcct\n",
      ">Horn_1;size=10\n",
      "tatagggccaacccgtctctgtggcaaaagagtgggacgagccccgagtagaggtgataa\n",
      "acctatcgagcctagttatagctggttgcttaggaaatgaatagaagttcagccccctgg\n",
      "ctttcttaaaaccccaaggtaaaactaaccttgtcccaaagaaaccaagagagttaatca\n",
      "aaggaggtacagctcct\n"
     ]
    }
   ],
   "source": [
    "!cat ./set1_1_derep.msa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "id": "ba4c747d-aa01-40c4-8822-5e0c26078198",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Horn_1;size=40\n",
      "ccaatcacccggtcttggttaaaatccaaggaaagataatgaacttgattacaacaatcc\n",
      "ttgctattactattacattgtccgcagtactagccactatttctttctgattaccacaaa\n",
      "tttcccccgacgcagaaaaactctccccctacgaatgtggatttgaccccctagga\n",
      ">Horn_10;size=4\n",
      "ccaatcacccggtcttggttaaaatccaaggaaagataatgaacttgattacaacaatcc\n",
      "ttgctattactattacattgtccacagtactagccactatttctttctgattaccacaaa\n",
      "tttcccccgacgcagaaaaactctccccctacgaatgtggatttgaccccctagga\n"
     ]
    }
   ],
   "source": [
    "!cat ./set2_1_derep.msa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 278,
   "id": "8acead6a-7db2-430f-966b-9fb4f97f5f22",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 206 ms, sys: 1.96 ms, total: 208 ms\n",
      "Wall time: 208 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "seeds = []\n",
    "for window in [i.split(\".\")[0] for i in glob.glob(\"set1_*.derep.uc\")]:\n",
    "    tmp = pd.read_csv(\"./\" + window + \".derep.uc\", sep = \"\\t\", header = None)\n",
    "    tmp = tmp[tmp[9] != \"*\"]\n",
    "    tmp[\"hit\"] = (tmp[8].str.replace(\"Assinica_\", \"A\").str.replace(\"Horn_\", \"H\").str.replace(\"Dom_\", \"D\")\n",
    "     .str.replace(\"LTup_\", \"L\").str.replace(\"Temi_\", \"T\").str.replace(\"Wind_\", \"W\").str.replace(\"Ass_\", \"A\"))\n",
    "    tmp[\"strain_id_hit\"] = (tmp[8].str.replace(\"Assinica_\", \"A\").str.replace(\"Horn_\", \"H\").str.replace(\"Dom_\", \"D\")\n",
    "     .str.replace(\"LTup_\", \"L\").str.replace(\"Temi_\", \"T\").str.replace(\"Wind_\", \"W\").str.replace(\"Ass_\", \"A\").str.split(\"\", expand =True))[1]\n",
    "    tmp[\"seed\"] = (tmp[9].str.replace(\"Assinica_\", \"A\").str.replace(\"Horn_\", \"H\").str.replace(\"Dom_\", \"D\")\n",
    "     .str.replace(\"LTup_\", \"L\").str.replace(\"Temi_\", \"T\").str.replace(\"Wind_\", \"W\").str.replace(\"Ass_\", \"A\"))\n",
    "    tmp[\"strain_id_seed\"] = (tmp[9].str.replace(\"Assinica_\", \"A\").str.replace(\"Horn_\", \"H\").str.replace(\"Dom_\", \"D\")\n",
    "     .str.replace(\"LTup_\", \"L\").str.replace(\"Temi_\", \"T\").str.replace(\"Wind_\", \"W\").str.replace(\"Ass_\", \"A\").str.split(\"\", expand =True))[1]\n",
    "    for i in tmp[9].unique():\n",
    "        tmp1 = tmp[tmp[9] == i].reset_index(drop=True)\n",
    "        tmplist = [tmp1[\"seed\"][0]] + tmp1[\"hit\"].tolist()\n",
    "        tmpstrains = list(set([tmp1[\"strain_id_seed\"][0]] + tmp1[\"strain_id_hit\"].tolist()))\n",
    "        seeds = seeds + [[window, i, tmp1[\"seed\"][0], \"|\".join(tmplist), tmp1[\"strain_id_seed\"][0], \"|\".join(tmpstrains)]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 279,
   "id": "cab3e9a0-250d-4816-9a63-a0d78ab3cd6e",
   "metadata": {},
   "outputs": [],
   "source": [
    "seeds = pd.DataFrame(seeds)\n",
    "seeds[\"start\"] = seeds[0].str.split(\"_\", expand = True)[1].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "id": "d0c2f179-6e70-4a34-91c3-6b993e87ac40",
   "metadata": {},
   "outputs": [],
   "source": [
    "mask = (seeds[5].str.len() == 1)\n",
    "tmp = seeds.loc[mask].sort_values(by=3, key=lambda x: x.str.len(), ascending = False).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 281,
   "id": "a322ffc2-1e45-401d-a38c-8f817e858a54",
   "metadata": {},
   "outputs": [],
   "source": [
    "import itertools\n",
    "merged = list(set(list(itertools.chain(*[i.split(\"|\") for i in seeds[3].tolist()]))))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 282,
   "id": "3e5d8055-e872-479b-a008-c4044d9b151d",
   "metadata": {},
   "outputs": [],
   "source": [
    "H = [i for i in merged if \"H\" in i]\n",
    "W = [i for i in merged if \"W\" in i]\n",
    "T = [i for i in merged if \"T\" in i]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 283,
   "id": "0c605c19-0448-4ba5-aeb4-b834ed6a1610",
   "metadata": {},
   "outputs": [],
   "source": [
    "informative = pd.DataFrame()\n",
    "for i in tmp[3].unique():\n",
    "    test = tmp[tmp[0].isin(tmp[0][tmp[3] == i].tolist())]\n",
    "    for k in test[0][test.duplicated(0, keep=False) == True].tolist():\n",
    "        if len(test[test[0] == k]) > 0:\n",
    "            informative = pd.concat([informative, pd.DataFrame(data = [i, k, len(test[test[0] == k]), \",\".join(list(set(test[3][test[0] == k].tolist())))]).T]).reset_index(drop=True)\n",
    "    if len(test[0][test.duplicated(0, keep=False) == True].tolist()) < 1:\n",
    "        test1 = test.copy().reset_index(drop=True)\n",
    "        for k in range(len(test1)):\n",
    "            informative = pd.concat([informative, pd.DataFrame(data = [i, test1[0][k], 1, i]).T]).reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "id": "d8acbb77-2b0b-414d-be02-9e8f7b3b92d3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "T\n",
      "T\n",
      "T\n"
     ]
    }
   ],
   "source": [
    "for seq_record in SeqIO.parse(\"set1_1_derep.fasta\", \"fasta\"):\n",
    "    print(seq_record.seq[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 303,
   "id": "e6a86663-5030-4dc4-8754-b4729b665220",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/1966945972.py:22: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  set1[\"Variant Positions (1-based)\"][pair] = str([i+1 for i in variant_positions])\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/1966945972.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  set1[\"Variant Positions (1-based)\"][pair] = str([i+1 for i in variant_positions])\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/1966945972.py:22: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[23, 130]' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  set1[\"Variant Positions (1-based)\"][pair] = str([i+1 for i in variant_positions])\n"
     ]
    }
   ],
   "source": [
    "set1[\"Variant Positions (1-based)\"] = np.nan\n",
    "for pair in range(len(set1)):\n",
    "    # Load sequences from the FASTA file\n",
    "    fasta_file = \"set1_\" + str(set1[\"Primer pair #\"][pair]) + \"_derep.fasta\"\n",
    "    sequences = list(SeqIO.parse(fasta_file, \"fasta\"))\n",
    "    \n",
    "    # Check that all sequences are the same length\n",
    "    seq_lengths = set(len(seq.seq) for seq in sequences)\n",
    "    if len(seq_lengths) != 1:\n",
    "        raise ValueError(\"Sequences are not all the same length!\")\n",
    "    \n",
    "    # Find variant positions\n",
    "    variant_positions = []\n",
    "    sequence_length = len(sequences[0].seq)\n",
    "    \n",
    "    for i in range(sequence_length):\n",
    "        bases = set(seq.seq[i] for seq in sequences)\n",
    "        if len(bases) > 1:\n",
    "            variant_positions.append(i)\n",
    "    \n",
    "    # Output the results\n",
    "    set1[\"Variant Positions (1-based)\"][pair] = str([i+1 for i in variant_positions])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "id": "f0e54bad-d222-4f3b-be66-437c84b70e34",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/2650099908.py:22: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!\n",
      "You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.\n",
      "A typical example is when you are setting values in a column of a DataFrame, like:\n",
      "\n",
      "df[\"col\"][row_indexer] = value\n",
      "\n",
      "Use `df.loc[row_indexer, \"col\"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "\n",
      "  set2[\"Variant Positions (1-based)\"][pair] = str([i+1 for i in variant_positions])\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/2650099908.py:22: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  set2[\"Variant Positions (1-based)\"][pair] = str([i+1 for i in variant_positions])\n",
      "/fs/dss/scratch/7292286_noge4093_mpcs055/ipykernel_3625234/2650099908.py:22: FutureWarning: Setting an item of incompatible dtype is deprecated and will raise an error in a future version of pandas. Value '[84]' has dtype incompatible with float64, please explicitly cast to a compatible dtype first.\n",
      "  set2[\"Variant Positions (1-based)\"][pair] = str([i+1 for i in variant_positions])\n"
     ]
    }
   ],
   "source": [
    "set2[\"Variant Positions (1-based)\"] = np.nan\n",
    "for pair in range(len(set2)):\n",
    "    # Load sequences from the FASTA file\n",
    "    fasta_file = \"set2_\" + str(set2[\"Primer pair #\"][pair]) + \"_derep.fasta\"\n",
    "    sequences = list(SeqIO.parse(fasta_file, \"fasta\"))\n",
    "    \n",
    "    # Check that all sequences are the same length\n",
    "    seq_lengths = set(len(seq.seq) for seq in sequences)\n",
    "    if len(seq_lengths) != 1:\n",
    "        raise ValueError(\"Sequences are not all the same length!\")\n",
    "    \n",
    "    # Find variant positions\n",
    "    variant_positions = []\n",
    "    sequence_length = len(sequences[0].seq)\n",
    "    \n",
    "    for i in range(sequence_length):\n",
    "        bases = set(seq.seq[i] for seq in sequences)\n",
    "        if len(bases) > 1:\n",
    "            variant_positions.append(i)\n",
    "    \n",
    "    # Output the results\n",
    "    set2[\"Variant Positions (1-based)\"][pair] = str([i+1 for i in variant_positions])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "id": "3b20317d-24a5-4667-8bf6-7cb1b6aab719",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Primer pair #</th>\n",
       "      <th>Forward primer Sequence (5'-&gt;3')</th>\n",
       "      <th>Forward primer Template strand</th>\n",
       "      <th>Forward primer Length</th>\n",
       "      <th>Forward primer Start</th>\n",
       "      <th>Forward primer Stop</th>\n",
       "      <th>Forward primer Tm</th>\n",
       "      <th>Forward primer GC%</th>\n",
       "      <th>Forward primer Self complementarity</th>\n",
       "      <th>Forward primer Self 3' complementarity</th>\n",
       "      <th>...</th>\n",
       "      <th>Reverse primer Length</th>\n",
       "      <th>Reverse primer Start</th>\n",
       "      <th>Reverse primer Stop</th>\n",
       "      <th>Reverse primer Tm</th>\n",
       "      <th>Reverse primer GC%</th>\n",
       "      <th>Reverse primer Self complementarity</th>\n",
       "      <th>Reverse primer Self 3' complementarity</th>\n",
       "      <th>Product length</th>\n",
       "      <th>Unnamed: 20</th>\n",
       "      <th>Variant Positions (1-based)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>TATAGGGCCAACCCGTCTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2353</td>\n",
       "      <td>2372</td>\n",
       "      <td>59.74</td>\n",
       "      <td>55.00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>2549</td>\n",
       "      <td>2530</td>\n",
       "      <td>57.67</td>\n",
       "      <td>50.00</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[23, 130]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>ACTAGACGAGCTACTCCGGG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2324</td>\n",
       "      <td>2343</td>\n",
       "      <td>60.18</td>\n",
       "      <td>60.00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2521</td>\n",
       "      <td>2500</td>\n",
       "      <td>58.97</td>\n",
       "      <td>45.45</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[52, 159]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>TAGACGAGCTACTCCGGGAC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2326</td>\n",
       "      <td>2345</td>\n",
       "      <td>60.18</td>\n",
       "      <td>60.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>2519</td>\n",
       "      <td>2499</td>\n",
       "      <td>57.89</td>\n",
       "      <td>42.86</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>194</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[50, 157]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>ATAGGGCCAACCCGTCTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>2354</td>\n",
       "      <td>2372</td>\n",
       "      <td>60.00</td>\n",
       "      <td>57.89</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>2519</td>\n",
       "      <td>2500</td>\n",
       "      <td>57.16</td>\n",
       "      <td>45.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>166</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[22, 129]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>GCTACTCCGGGACAGCCTAT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2333</td>\n",
       "      <td>2352</td>\n",
       "      <td>60.83</td>\n",
       "      <td>60.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>2520</td>\n",
       "      <td>2500</td>\n",
       "      <td>57.89</td>\n",
       "      <td>42.86</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>188</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[43, 150]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>TTATAGGGCCAACCCGTCTC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2352</td>\n",
       "      <td>2371</td>\n",
       "      <td>58.87</td>\n",
       "      <td>55.00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>2549</td>\n",
       "      <td>2529</td>\n",
       "      <td>58.16</td>\n",
       "      <td>47.62</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[24, 131]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>ATAGGGCCAACCCGTCTCTG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2354</td>\n",
       "      <td>2373</td>\n",
       "      <td>61.34</td>\n",
       "      <td>60.00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2549</td>\n",
       "      <td>2528</td>\n",
       "      <td>58.81</td>\n",
       "      <td>45.45</td>\n",
       "      <td>8.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>196</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[22, 129]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>TTATAGGGCCAACCCGTCTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>2352</td>\n",
       "      <td>2372</td>\n",
       "      <td>60.34</td>\n",
       "      <td>52.38</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2526</td>\n",
       "      <td>2504</td>\n",
       "      <td>59.80</td>\n",
       "      <td>43.48</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>175</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[24, 131]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>AGACGAGCTACTCCGGGAC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>2327</td>\n",
       "      <td>2345</td>\n",
       "      <td>60.45</td>\n",
       "      <td>63.16</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2522</td>\n",
       "      <td>2501</td>\n",
       "      <td>58.70</td>\n",
       "      <td>45.45</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>196</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[49, 156]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>GCTACTCCGGGACAGCCTA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>2333</td>\n",
       "      <td>2351</td>\n",
       "      <td>60.46</td>\n",
       "      <td>63.16</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2522</td>\n",
       "      <td>2500</td>\n",
       "      <td>60.31</td>\n",
       "      <td>43.48</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[43, 150]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>ATTATAGGGCCAACCCGTCTC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>2351</td>\n",
       "      <td>2371</td>\n",
       "      <td>59.30</td>\n",
       "      <td>52.38</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2523</td>\n",
       "      <td>2501</td>\n",
       "      <td>59.68</td>\n",
       "      <td>47.83</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>173</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[25, 132]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>CTACTCCGGGACAGCCTATT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2334</td>\n",
       "      <td>2353</td>\n",
       "      <td>58.29</td>\n",
       "      <td>55.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2520</td>\n",
       "      <td>2499</td>\n",
       "      <td>58.55</td>\n",
       "      <td>40.91</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>187</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[42, 149]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13</td>\n",
       "      <td>GACGAGCTACTCCGGGACAG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2328</td>\n",
       "      <td>2347</td>\n",
       "      <td>61.72</td>\n",
       "      <td>65.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2521</td>\n",
       "      <td>2499</td>\n",
       "      <td>59.55</td>\n",
       "      <td>43.48</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>194</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[48, 155]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>ATTATAGGGCCAACCCGTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2351</td>\n",
       "      <td>2370</td>\n",
       "      <td>58.18</td>\n",
       "      <td>50.00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>2548</td>\n",
       "      <td>2528</td>\n",
       "      <td>57.36</td>\n",
       "      <td>47.62</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[25, 132]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>GACGAGCTACTCCGGGACA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>2328</td>\n",
       "      <td>2346</td>\n",
       "      <td>60.75</td>\n",
       "      <td>63.16</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2525</td>\n",
       "      <td>2504</td>\n",
       "      <td>58.18</td>\n",
       "      <td>45.45</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[48, 155]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16</td>\n",
       "      <td>ACGAGCTACTCCGGGACAG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>2329</td>\n",
       "      <td>2347</td>\n",
       "      <td>60.75</td>\n",
       "      <td>63.16</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2526</td>\n",
       "      <td>2505</td>\n",
       "      <td>58.18</td>\n",
       "      <td>45.45</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[47, 154]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>AACTAGACGAGCTACTCCGGG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>2323</td>\n",
       "      <td>2343</td>\n",
       "      <td>60.75</td>\n",
       "      <td>57.14</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2519</td>\n",
       "      <td>2497</td>\n",
       "      <td>58.79</td>\n",
       "      <td>43.48</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>197</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[53, 160]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>ACGAGCTACTCCGGGACA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>18</td>\n",
       "      <td>2329</td>\n",
       "      <td>2346</td>\n",
       "      <td>59.65</td>\n",
       "      <td>61.11</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>2521</td>\n",
       "      <td>2501</td>\n",
       "      <td>57.27</td>\n",
       "      <td>47.62</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>193</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[47, 154]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>TAGGGCCAACCCGTCTCTG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>2355</td>\n",
       "      <td>2373</td>\n",
       "      <td>60.99</td>\n",
       "      <td>63.16</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2524</td>\n",
       "      <td>2502</td>\n",
       "      <td>58.79</td>\n",
       "      <td>43.48</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>170</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[21, 128]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20</td>\n",
       "      <td>AGCTACTCCGGGACAGCCTA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2332</td>\n",
       "      <td>2351</td>\n",
       "      <td>61.93</td>\n",
       "      <td>60.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2525</td>\n",
       "      <td>2502</td>\n",
       "      <td>59.72</td>\n",
       "      <td>45.83</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>194</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[44, 151]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>21</td>\n",
       "      <td>TATAGGGCCAACCCGTCTCTG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>2353</td>\n",
       "      <td>2373</td>\n",
       "      <td>61.03</td>\n",
       "      <td>57.14</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2525</td>\n",
       "      <td>2503</td>\n",
       "      <td>58.79</td>\n",
       "      <td>43.48</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>173</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[23, 130]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>CCGAAACTAGACGAGCTACTCC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>2319</td>\n",
       "      <td>2340</td>\n",
       "      <td>59.97</td>\n",
       "      <td>54.55</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2518</td>\n",
       "      <td>2496</td>\n",
       "      <td>58.79</td>\n",
       "      <td>43.48</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[57, 164]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23</td>\n",
       "      <td>ACTCCGGGACAGCCTATTAT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2336</td>\n",
       "      <td>2355</td>\n",
       "      <td>57.60</td>\n",
       "      <td>50.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>2524</td>\n",
       "      <td>2504</td>\n",
       "      <td>57.06</td>\n",
       "      <td>42.86</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>189</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[40, 147]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24</td>\n",
       "      <td>CTAGACGAGCTACTCCGGGAC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>2325</td>\n",
       "      <td>2345</td>\n",
       "      <td>61.14</td>\n",
       "      <td>61.90</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2519</td>\n",
       "      <td>2496</td>\n",
       "      <td>60.33</td>\n",
       "      <td>41.67</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>195</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[51, 158]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25</td>\n",
       "      <td>AGACGAGCTACTCCGGGACA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2327</td>\n",
       "      <td>2346</td>\n",
       "      <td>62.19</td>\n",
       "      <td>60.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2526</td>\n",
       "      <td>2503</td>\n",
       "      <td>60.33</td>\n",
       "      <td>41.67</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[49, 156]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26</td>\n",
       "      <td>CGAAACTAGACGAGCTACTCCG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>2320</td>\n",
       "      <td>2341</td>\n",
       "      <td>60.29</td>\n",
       "      <td>54.55</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2519</td>\n",
       "      <td>2498</td>\n",
       "      <td>57.75</td>\n",
       "      <td>40.91</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[56, 163]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>27</td>\n",
       "      <td>GGCCAACCCGTCTCTGTG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>18</td>\n",
       "      <td>2358</td>\n",
       "      <td>2375</td>\n",
       "      <td>60.36</td>\n",
       "      <td>66.67</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2527</td>\n",
       "      <td>2505</td>\n",
       "      <td>58.79</td>\n",
       "      <td>43.48</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>170</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[125]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28</td>\n",
       "      <td>TAGGGCCAACCCGTCTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>18</td>\n",
       "      <td>2355</td>\n",
       "      <td>2372</td>\n",
       "      <td>59.55</td>\n",
       "      <td>61.11</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2524</td>\n",
       "      <td>2503</td>\n",
       "      <td>57.76</td>\n",
       "      <td>40.91</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>170</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[21, 128]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>29</td>\n",
       "      <td>GCTACTCCGGGACAGCCTATT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>2333</td>\n",
       "      <td>2353</td>\n",
       "      <td>61.37</td>\n",
       "      <td>57.14</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2527</td>\n",
       "      <td>2504</td>\n",
       "      <td>60.33</td>\n",
       "      <td>41.67</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>195</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[43, 150]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>30</td>\n",
       "      <td>CCCCGAAACTAGACGAGCTAC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>2317</td>\n",
       "      <td>2337</td>\n",
       "      <td>59.93</td>\n",
       "      <td>57.14</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>2514</td>\n",
       "      <td>2490</td>\n",
       "      <td>59.17</td>\n",
       "      <td>40.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[59, 166]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>31</td>\n",
       "      <td>ATTATAGGGCCAACCCGTCTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>2351</td>\n",
       "      <td>2372</td>\n",
       "      <td>60.70</td>\n",
       "      <td>50.00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2549</td>\n",
       "      <td>2527</td>\n",
       "      <td>58.64</td>\n",
       "      <td>43.48</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>199</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[25, 132]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>32</td>\n",
       "      <td>CTAGACGAGCTACTCCGGG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>2325</td>\n",
       "      <td>2343</td>\n",
       "      <td>58.38</td>\n",
       "      <td>63.16</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2523</td>\n",
       "      <td>2502</td>\n",
       "      <td>57.43</td>\n",
       "      <td>45.45</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>199</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[51, 158]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>33</td>\n",
       "      <td>CCCCCGAAACTAGACGAGC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>2316</td>\n",
       "      <td>2334</td>\n",
       "      <td>59.86</td>\n",
       "      <td>63.16</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2513</td>\n",
       "      <td>2490</td>\n",
       "      <td>57.92</td>\n",
       "      <td>41.67</td>\n",
       "      <td>3.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[60, 167]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>34</td>\n",
       "      <td>TATAGGGCCAACCCGTCTC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>2353</td>\n",
       "      <td>2371</td>\n",
       "      <td>58.17</td>\n",
       "      <td>57.89</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2527</td>\n",
       "      <td>2506</td>\n",
       "      <td>57.48</td>\n",
       "      <td>40.91</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>175</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[23, 130]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>35</td>\n",
       "      <td>GCTACTCCGGGACAGCCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>18</td>\n",
       "      <td>2333</td>\n",
       "      <td>2350</td>\n",
       "      <td>60.76</td>\n",
       "      <td>66.67</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2520</td>\n",
       "      <td>2498</td>\n",
       "      <td>58.39</td>\n",
       "      <td>39.13</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>188</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[43, 150]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>36</td>\n",
       "      <td>AGCTACTCCGGGACAGCC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>18</td>\n",
       "      <td>2332</td>\n",
       "      <td>2349</td>\n",
       "      <td>60.76</td>\n",
       "      <td>66.67</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2520</td>\n",
       "      <td>2497</td>\n",
       "      <td>59.35</td>\n",
       "      <td>41.67</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>189</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[44, 151]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>37</td>\n",
       "      <td>GAGCTACTCCGGGACAGCC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>2331</td>\n",
       "      <td>2349</td>\n",
       "      <td>61.79</td>\n",
       "      <td>68.42</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2521</td>\n",
       "      <td>2498</td>\n",
       "      <td>59.35</td>\n",
       "      <td>41.67</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>191</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[45, 152]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>38</td>\n",
       "      <td>TAGACGAGCTACTCCGGGACA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>2326</td>\n",
       "      <td>2346</td>\n",
       "      <td>61.85</td>\n",
       "      <td>57.14</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2518</td>\n",
       "      <td>2495</td>\n",
       "      <td>59.35</td>\n",
       "      <td>41.67</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>193</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[50, 157]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>39</td>\n",
       "      <td>TAGGGCCAACCCGTCTCTGT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2355</td>\n",
       "      <td>2374</td>\n",
       "      <td>62.74</td>\n",
       "      <td>60.00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2522</td>\n",
       "      <td>2499</td>\n",
       "      <td>60.82</td>\n",
       "      <td>41.67</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>168</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[21, 128]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>40</td>\n",
       "      <td>TATTATAGGGCCAACCCGTCTC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>2350</td>\n",
       "      <td>2371</td>\n",
       "      <td>59.10</td>\n",
       "      <td>50.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2548</td>\n",
       "      <td>2527</td>\n",
       "      <td>57.24</td>\n",
       "      <td>45.45</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>199</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[26, 133]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>41</td>\n",
       "      <td>CTACTCCGGGACAGCCTATTA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>2334</td>\n",
       "      <td>2354</td>\n",
       "      <td>58.13</td>\n",
       "      <td>52.38</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>2518</td>\n",
       "      <td>2497</td>\n",
       "      <td>57.15</td>\n",
       "      <td>45.45</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[42, 149]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>42</td>\n",
       "      <td>CTATTATAGGGCCAACCCGTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>2349</td>\n",
       "      <td>2370</td>\n",
       "      <td>59.10</td>\n",
       "      <td>50.00</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2548</td>\n",
       "      <td>2525</td>\n",
       "      <td>59.12</td>\n",
       "      <td>45.83</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[27, 134]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>43</td>\n",
       "      <td>CCCGAAACTAGACGAGCTACT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>2318</td>\n",
       "      <td>2338</td>\n",
       "      <td>58.98</td>\n",
       "      <td>52.38</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>2517</td>\n",
       "      <td>2495</td>\n",
       "      <td>57.15</td>\n",
       "      <td>39.13</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[58, 165]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>44</td>\n",
       "      <td>TATTATAGGGCCAACCCGTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>2350</td>\n",
       "      <td>2370</td>\n",
       "      <td>58.02</td>\n",
       "      <td>47.62</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2524</td>\n",
       "      <td>2501</td>\n",
       "      <td>60.93</td>\n",
       "      <td>45.83</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>175</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[26, 133]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>45</td>\n",
       "      <td>GGCCAACCCGTCTCTGT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>17</td>\n",
       "      <td>2358</td>\n",
       "      <td>2374</td>\n",
       "      <td>58.84</td>\n",
       "      <td>64.71</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>2557</td>\n",
       "      <td>2537</td>\n",
       "      <td>57.14</td>\n",
       "      <td>42.86</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[18, 125]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>46</td>\n",
       "      <td>GCCCCCGAAACTAGACGAG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>2315</td>\n",
       "      <td>2333</td>\n",
       "      <td>59.86</td>\n",
       "      <td>63.16</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2514</td>\n",
       "      <td>2491</td>\n",
       "      <td>57.05</td>\n",
       "      <td>37.50</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[61, 168]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>47</td>\n",
       "      <td>CTATTATAGGGCCAACCCGTCTC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>23</td>\n",
       "      <td>2349</td>\n",
       "      <td>2371</td>\n",
       "      <td>60.06</td>\n",
       "      <td>52.17</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>2528</td>\n",
       "      <td>2504</td>\n",
       "      <td>60.10</td>\n",
       "      <td>40.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[27, 134]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>48</td>\n",
       "      <td>GCTACTCCGGGACAGCCTATTA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>2333</td>\n",
       "      <td>2354</td>\n",
       "      <td>61.08</td>\n",
       "      <td>54.55</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>2518</td>\n",
       "      <td>2494</td>\n",
       "      <td>59.87</td>\n",
       "      <td>40.00</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>186</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[43, 150]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>49</td>\n",
       "      <td>CCCCGAAACTAGACGAGCTA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>2317</td>\n",
       "      <td>2336</td>\n",
       "      <td>58.61</td>\n",
       "      <td>55.00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2516</td>\n",
       "      <td>2493</td>\n",
       "      <td>57.14</td>\n",
       "      <td>33.33</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[59, 166]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>50</td>\n",
       "      <td>GACGAGCTACTCCGGGAC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>18</td>\n",
       "      <td>2328</td>\n",
       "      <td>2345</td>\n",
       "      <td>58.89</td>\n",
       "      <td>66.67</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>2523</td>\n",
       "      <td>2500</td>\n",
       "      <td>61.17</td>\n",
       "      <td>45.83</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>196</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[48, 155]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>50 rows  22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Primer pair # Forward primer Sequence (5'->3')  \\\n",
       "0               1             TATAGGGCCAACCCGTCTCT   \n",
       "1               2             ACTAGACGAGCTACTCCGGG   \n",
       "2               3             TAGACGAGCTACTCCGGGAC   \n",
       "3               4              ATAGGGCCAACCCGTCTCT   \n",
       "4               5             GCTACTCCGGGACAGCCTAT   \n",
       "5               6             TTATAGGGCCAACCCGTCTC   \n",
       "6               7             ATAGGGCCAACCCGTCTCTG   \n",
       "7               8            TTATAGGGCCAACCCGTCTCT   \n",
       "8               9              AGACGAGCTACTCCGGGAC   \n",
       "9              10              GCTACTCCGGGACAGCCTA   \n",
       "10             11            ATTATAGGGCCAACCCGTCTC   \n",
       "11             12             CTACTCCGGGACAGCCTATT   \n",
       "12             13             GACGAGCTACTCCGGGACAG   \n",
       "13             14             ATTATAGGGCCAACCCGTCT   \n",
       "14             15              GACGAGCTACTCCGGGACA   \n",
       "15             16              ACGAGCTACTCCGGGACAG   \n",
       "16             17            AACTAGACGAGCTACTCCGGG   \n",
       "17             18               ACGAGCTACTCCGGGACA   \n",
       "18             19              TAGGGCCAACCCGTCTCTG   \n",
       "19             20             AGCTACTCCGGGACAGCCTA   \n",
       "20             21            TATAGGGCCAACCCGTCTCTG   \n",
       "21             22           CCGAAACTAGACGAGCTACTCC   \n",
       "22             23             ACTCCGGGACAGCCTATTAT   \n",
       "23             24            CTAGACGAGCTACTCCGGGAC   \n",
       "24             25             AGACGAGCTACTCCGGGACA   \n",
       "25             26           CGAAACTAGACGAGCTACTCCG   \n",
       "26             27               GGCCAACCCGTCTCTGTG   \n",
       "27             28               TAGGGCCAACCCGTCTCT   \n",
       "28             29            GCTACTCCGGGACAGCCTATT   \n",
       "29             30            CCCCGAAACTAGACGAGCTAC   \n",
       "30             31           ATTATAGGGCCAACCCGTCTCT   \n",
       "31             32              CTAGACGAGCTACTCCGGG   \n",
       "32             33              CCCCCGAAACTAGACGAGC   \n",
       "33             34              TATAGGGCCAACCCGTCTC   \n",
       "34             35               GCTACTCCGGGACAGCCT   \n",
       "35             36               AGCTACTCCGGGACAGCC   \n",
       "36             37              GAGCTACTCCGGGACAGCC   \n",
       "37             38            TAGACGAGCTACTCCGGGACA   \n",
       "38             39             TAGGGCCAACCCGTCTCTGT   \n",
       "39             40           TATTATAGGGCCAACCCGTCTC   \n",
       "40             41            CTACTCCGGGACAGCCTATTA   \n",
       "41             42           CTATTATAGGGCCAACCCGTCT   \n",
       "42             43            CCCGAAACTAGACGAGCTACT   \n",
       "43             44            TATTATAGGGCCAACCCGTCT   \n",
       "44             45                GGCCAACCCGTCTCTGT   \n",
       "45             46              GCCCCCGAAACTAGACGAG   \n",
       "46             47          CTATTATAGGGCCAACCCGTCTC   \n",
       "47             48           GCTACTCCGGGACAGCCTATTA   \n",
       "48             49             CCCCGAAACTAGACGAGCTA   \n",
       "49             50               GACGAGCTACTCCGGGAC   \n",
       "\n",
       "   Forward primer Template strand  Forward primer Length  \\\n",
       "0                            Plus                     20   \n",
       "1                            Plus                     20   \n",
       "2                            Plus                     20   \n",
       "3                            Plus                     19   \n",
       "4                            Plus                     20   \n",
       "5                            Plus                     20   \n",
       "6                            Plus                     20   \n",
       "7                            Plus                     21   \n",
       "8                            Plus                     19   \n",
       "9                            Plus                     19   \n",
       "10                           Plus                     21   \n",
       "11                           Plus                     20   \n",
       "12                           Plus                     20   \n",
       "13                           Plus                     20   \n",
       "14                           Plus                     19   \n",
       "15                           Plus                     19   \n",
       "16                           Plus                     21   \n",
       "17                           Plus                     18   \n",
       "18                           Plus                     19   \n",
       "19                           Plus                     20   \n",
       "20                           Plus                     21   \n",
       "21                           Plus                     22   \n",
       "22                           Plus                     20   \n",
       "23                           Plus                     21   \n",
       "24                           Plus                     20   \n",
       "25                           Plus                     22   \n",
       "26                           Plus                     18   \n",
       "27                           Plus                     18   \n",
       "28                           Plus                     21   \n",
       "29                           Plus                     21   \n",
       "30                           Plus                     22   \n",
       "31                           Plus                     19   \n",
       "32                           Plus                     19   \n",
       "33                           Plus                     19   \n",
       "34                           Plus                     18   \n",
       "35                           Plus                     18   \n",
       "36                           Plus                     19   \n",
       "37                           Plus                     21   \n",
       "38                           Plus                     20   \n",
       "39                           Plus                     22   \n",
       "40                           Plus                     21   \n",
       "41                           Plus                     22   \n",
       "42                           Plus                     21   \n",
       "43                           Plus                     21   \n",
       "44                           Plus                     17   \n",
       "45                           Plus                     19   \n",
       "46                           Plus                     23   \n",
       "47                           Plus                     22   \n",
       "48                           Plus                     20   \n",
       "49                           Plus                     18   \n",
       "\n",
       "    Forward primer Start  Forward primer Stop  Forward primer Tm  \\\n",
       "0                   2353                 2372              59.74   \n",
       "1                   2324                 2343              60.18   \n",
       "2                   2326                 2345              60.18   \n",
       "3                   2354                 2372              60.00   \n",
       "4                   2333                 2352              60.83   \n",
       "5                   2352                 2371              58.87   \n",
       "6                   2354                 2373              61.34   \n",
       "7                   2352                 2372              60.34   \n",
       "8                   2327                 2345              60.45   \n",
       "9                   2333                 2351              60.46   \n",
       "10                  2351                 2371              59.30   \n",
       "11                  2334                 2353              58.29   \n",
       "12                  2328                 2347              61.72   \n",
       "13                  2351                 2370              58.18   \n",
       "14                  2328                 2346              60.75   \n",
       "15                  2329                 2347              60.75   \n",
       "16                  2323                 2343              60.75   \n",
       "17                  2329                 2346              59.65   \n",
       "18                  2355                 2373              60.99   \n",
       "19                  2332                 2351              61.93   \n",
       "20                  2353                 2373              61.03   \n",
       "21                  2319                 2340              59.97   \n",
       "22                  2336                 2355              57.60   \n",
       "23                  2325                 2345              61.14   \n",
       "24                  2327                 2346              62.19   \n",
       "25                  2320                 2341              60.29   \n",
       "26                  2358                 2375              60.36   \n",
       "27                  2355                 2372              59.55   \n",
       "28                  2333                 2353              61.37   \n",
       "29                  2317                 2337              59.93   \n",
       "30                  2351                 2372              60.70   \n",
       "31                  2325                 2343              58.38   \n",
       "32                  2316                 2334              59.86   \n",
       "33                  2353                 2371              58.17   \n",
       "34                  2333                 2350              60.76   \n",
       "35                  2332                 2349              60.76   \n",
       "36                  2331                 2349              61.79   \n",
       "37                  2326                 2346              61.85   \n",
       "38                  2355                 2374              62.74   \n",
       "39                  2350                 2371              59.10   \n",
       "40                  2334                 2354              58.13   \n",
       "41                  2349                 2370              59.10   \n",
       "42                  2318                 2338              58.98   \n",
       "43                  2350                 2370              58.02   \n",
       "44                  2358                 2374              58.84   \n",
       "45                  2315                 2333              59.86   \n",
       "46                  2349                 2371              60.06   \n",
       "47                  2333                 2354              61.08   \n",
       "48                  2317                 2336              58.61   \n",
       "49                  2328                 2345              58.89   \n",
       "\n",
       "    Forward primer GC%  Forward primer Self complementarity  \\\n",
       "0                55.00                                  4.0   \n",
       "1                60.00                                  4.0   \n",
       "2                60.00                                  5.0   \n",
       "3                57.89                                  4.0   \n",
       "4                60.00                                  5.0   \n",
       "5                55.00                                  4.0   \n",
       "6                60.00                                  4.0   \n",
       "7                52.38                                  4.0   \n",
       "8                63.16                                  5.0   \n",
       "9                63.16                                  5.0   \n",
       "10               52.38                                  4.0   \n",
       "11               55.00                                  5.0   \n",
       "12               65.00                                  5.0   \n",
       "13               50.00                                  4.0   \n",
       "14               63.16                                  5.0   \n",
       "15               63.16                                  5.0   \n",
       "16               57.14                                  4.0   \n",
       "17               61.11                                  5.0   \n",
       "18               63.16                                  4.0   \n",
       "19               60.00                                  5.0   \n",
       "20               57.14                                  4.0   \n",
       "21               54.55                                  4.0   \n",
       "22               50.00                                  5.0   \n",
       "23               61.90                                  5.0   \n",
       "24               60.00                                  5.0   \n",
       "25               54.55                                  4.0   \n",
       "26               66.67                                  4.0   \n",
       "27               61.11                                  4.0   \n",
       "28               57.14                                  5.0   \n",
       "29               57.14                                  4.0   \n",
       "30               50.00                                  4.0   \n",
       "31               63.16                                  4.0   \n",
       "32               63.16                                  4.0   \n",
       "33               57.89                                  4.0   \n",
       "34               66.67                                  5.0   \n",
       "35               66.67                                  5.0   \n",
       "36               68.42                                  5.0   \n",
       "37               57.14                                  5.0   \n",
       "38               60.00                                  4.0   \n",
       "39               50.00                                  5.0   \n",
       "40               52.38                                  5.0   \n",
       "41               50.00                                  7.0   \n",
       "42               52.38                                  4.0   \n",
       "43               47.62                                  5.0   \n",
       "44               64.71                                  4.0   \n",
       "45               63.16                                  4.0   \n",
       "46               52.17                                  7.0   \n",
       "47               54.55                                  5.0   \n",
       "48               55.00                                  4.0   \n",
       "49               66.67                                  5.0   \n",
       "\n",
       "    Forward primer Self 3' complementarity  ... Reverse primer Length  \\\n",
       "0                                      0.0  ...                    20   \n",
       "1                                      3.0  ...                    22   \n",
       "2                                      3.0  ...                    21   \n",
       "3                                      0.0  ...                    20   \n",
       "4                                      3.0  ...                    21   \n",
       "5                                      0.0  ...                    21   \n",
       "6                                      1.0  ...                    22   \n",
       "7                                      0.0  ...                    23   \n",
       "8                                      3.0  ...                    22   \n",
       "9                                      2.0  ...                    23   \n",
       "10                                     0.0  ...                    23   \n",
       "11                                     1.0  ...                    22   \n",
       "12                                     3.0  ...                    23   \n",
       "13                                     0.0  ...                    21   \n",
       "14                                     1.0  ...                    22   \n",
       "15                                     3.0  ...                    22   \n",
       "16                                     3.0  ...                    23   \n",
       "17                                     1.0  ...                    21   \n",
       "18                                     1.0  ...                    23   \n",
       "19                                     2.0  ...                    24   \n",
       "20                                     1.0  ...                    23   \n",
       "21                                     2.0  ...                    23   \n",
       "22                                     3.0  ...                    21   \n",
       "23                                     3.0  ...                    24   \n",
       "24                                     1.0  ...                    24   \n",
       "25                                     2.0  ...                    22   \n",
       "26                                     1.0  ...                    23   \n",
       "27                                     0.0  ...                    22   \n",
       "28                                     1.0  ...                    24   \n",
       "29                                     3.0  ...                    25   \n",
       "30                                     0.0  ...                    23   \n",
       "31                                     3.0  ...                    22   \n",
       "32                                     2.0  ...                    24   \n",
       "33                                     0.0  ...                    22   \n",
       "34                                     3.0  ...                    23   \n",
       "35                                     3.0  ...                    24   \n",
       "36                                     3.0  ...                    24   \n",
       "37                                     1.0  ...                    24   \n",
       "38                                     0.0  ...                    24   \n",
       "39                                     0.0  ...                    22   \n",
       "40                                     3.0  ...                    22   \n",
       "41                                     0.0  ...                    24   \n",
       "42                                     2.0  ...                    23   \n",
       "43                                     0.0  ...                    24   \n",
       "44                                     0.0  ...                    21   \n",
       "45                                     3.0  ...                    24   \n",
       "46                                     0.0  ...                    25   \n",
       "47                                     3.0  ...                    25   \n",
       "48                                     2.0  ...                    24   \n",
       "49                                     3.0  ...                    24   \n",
       "\n",
       "   Reverse primer Start  Reverse primer Stop  Reverse primer Tm  \\\n",
       "0                  2549                 2530              57.67   \n",
       "1                  2521                 2500              58.97   \n",
       "2                  2519                 2499              57.89   \n",
       "3                  2519                 2500              57.16   \n",
       "4                  2520                 2500              57.89   \n",
       "5                  2549                 2529              58.16   \n",
       "6                  2549                 2528              58.81   \n",
       "7                  2526                 2504              59.80   \n",
       "8                  2522                 2501              58.70   \n",
       "9                  2522                 2500              60.31   \n",
       "10                 2523                 2501              59.68   \n",
       "11                 2520                 2499              58.55   \n",
       "12                 2521                 2499              59.55   \n",
       "13                 2548                 2528              57.36   \n",
       "14                 2525                 2504              58.18   \n",
       "15                 2526                 2505              58.18   \n",
       "16                 2519                 2497              58.79   \n",
       "17                 2521                 2501              57.27   \n",
       "18                 2524                 2502              58.79   \n",
       "19                 2525                 2502              59.72   \n",
       "20                 2525                 2503              58.79   \n",
       "21                 2518                 2496              58.79   \n",
       "22                 2524                 2504              57.06   \n",
       "23                 2519                 2496              60.33   \n",
       "24                 2526                 2503              60.33   \n",
       "25                 2519                 2498              57.75   \n",
       "26                 2527                 2505              58.79   \n",
       "27                 2524                 2503              57.76   \n",
       "28                 2527                 2504              60.33   \n",
       "29                 2514                 2490              59.17   \n",
       "30                 2549                 2527              58.64   \n",
       "31                 2523                 2502              57.43   \n",
       "32                 2513                 2490              57.92   \n",
       "33                 2527                 2506              57.48   \n",
       "34                 2520                 2498              58.39   \n",
       "35                 2520                 2497              59.35   \n",
       "36                 2521                 2498              59.35   \n",
       "37                 2518                 2495              59.35   \n",
       "38                 2522                 2499              60.82   \n",
       "39                 2548                 2527              57.24   \n",
       "40                 2518                 2497              57.15   \n",
       "41                 2548                 2525              59.12   \n",
       "42                 2517                 2495              57.15   \n",
       "43                 2524                 2501              60.93   \n",
       "44                 2557                 2537              57.14   \n",
       "45                 2514                 2491              57.05   \n",
       "46                 2528                 2504              60.10   \n",
       "47                 2518                 2494              59.87   \n",
       "48                 2516                 2493              57.14   \n",
       "49                 2523                 2500              61.17   \n",
       "\n",
       "    Reverse primer GC%  Reverse primer Self complementarity  \\\n",
       "0                50.00                                  8.0   \n",
       "1                45.45                                  4.0   \n",
       "2                42.86                                  3.0   \n",
       "3                45.00                                  3.0   \n",
       "4                42.86                                  3.0   \n",
       "5                47.62                                  8.0   \n",
       "6                45.45                                  8.0   \n",
       "7                43.48                                  2.0   \n",
       "8                45.45                                  4.0   \n",
       "9                43.48                                  4.0   \n",
       "10               47.83                                  4.0   \n",
       "11               40.91                                  3.0   \n",
       "12               43.48                                  4.0   \n",
       "13               47.62                                  6.0   \n",
       "14               45.45                                  2.0   \n",
       "15               45.45                                  2.0   \n",
       "16               43.48                                  3.0   \n",
       "17               47.62                                  4.0   \n",
       "18               43.48                                  4.0   \n",
       "19               45.83                                  4.0   \n",
       "20               43.48                                  3.0   \n",
       "21               43.48                                  3.0   \n",
       "22               42.86                                  2.0   \n",
       "23               41.67                                  3.0   \n",
       "24               41.67                                  3.0   \n",
       "25               40.91                                  3.0   \n",
       "26               43.48                                  3.0   \n",
       "27               40.91                                  3.0   \n",
       "28               41.67                                  3.0   \n",
       "29               40.00                                  3.0   \n",
       "30               43.48                                  8.0   \n",
       "31               45.45                                  4.0   \n",
       "32               41.67                                  3.0   \n",
       "33               40.91                                  3.0   \n",
       "34               39.13                                  3.0   \n",
       "35               41.67                                  3.0   \n",
       "36               41.67                                  4.0   \n",
       "37               41.67                                  3.0   \n",
       "38               41.67                                  4.0   \n",
       "39               45.45                                  6.0   \n",
       "40               45.45                                  3.0   \n",
       "41               45.83                                  6.0   \n",
       "42               39.13                                  3.0   \n",
       "43               45.83                                  4.0   \n",
       "44               42.86                                  4.0   \n",
       "45               37.50                                  3.0   \n",
       "46               40.00                                  3.0   \n",
       "47               40.00                                  3.0   \n",
       "48               33.33                                  3.0   \n",
       "49               45.83                                  4.0   \n",
       "\n",
       "    Reverse primer Self 3' complementarity  Product length  Unnamed: 20  \\\n",
       "0                                      1.0             197          NaN   \n",
       "1                                      1.0             198          NaN   \n",
       "2                                      2.0             194          NaN   \n",
       "3                                      1.0             166          NaN   \n",
       "4                                      1.0             188          NaN   \n",
       "5                                      2.0             198          NaN   \n",
       "6                                      1.0             196          NaN   \n",
       "7                                      1.0             175          NaN   \n",
       "8                                      3.0             196          NaN   \n",
       "9                                      1.0             190          NaN   \n",
       "10                                     3.0             173          NaN   \n",
       "11                                     2.0             187          NaN   \n",
       "12                                     2.0             194          NaN   \n",
       "13                                     1.0             198          NaN   \n",
       "14                                     1.0             198          NaN   \n",
       "15                                     1.0             198          NaN   \n",
       "16                                     0.0             197          NaN   \n",
       "17                                     3.0             193          NaN   \n",
       "18                                     1.0             170          NaN   \n",
       "19                                     1.0             194          NaN   \n",
       "20                                     3.0             173          NaN   \n",
       "21                                     2.0             200          NaN   \n",
       "22                                     1.0             189          NaN   \n",
       "23                                     2.0             195          NaN   \n",
       "24                                     3.0             200          NaN   \n",
       "25                                     2.0             200          NaN   \n",
       "26                                     1.0             170          NaN   \n",
       "27                                     3.0             170          NaN   \n",
       "28                                     1.0             195          NaN   \n",
       "29                                     3.0             198          NaN   \n",
       "30                                     2.0             199          NaN   \n",
       "31                                     1.0             199          NaN   \n",
       "32                                     3.0             198          NaN   \n",
       "33                                     0.0             175          NaN   \n",
       "34                                     2.0             188          NaN   \n",
       "35                                     0.0             189          NaN   \n",
       "36                                     2.0             191          NaN   \n",
       "37                                     0.0             193          NaN   \n",
       "38                                     2.0             168          NaN   \n",
       "39                                     2.0             199          NaN   \n",
       "40                                     0.0             185          NaN   \n",
       "41                                     2.0             200          NaN   \n",
       "42                                     0.0             200          NaN   \n",
       "43                                     3.0             175          NaN   \n",
       "44                                     3.0             200          NaN   \n",
       "45                                     1.0             200          NaN   \n",
       "46                                     1.0             180          NaN   \n",
       "47                                     1.0             186          NaN   \n",
       "48                                     0.0             200          NaN   \n",
       "49                                     1.0             196          NaN   \n",
       "\n",
       "    Variant Positions (1-based)  \n",
       "0                     [23, 130]  \n",
       "1                     [52, 159]  \n",
       "2                     [50, 157]  \n",
       "3                     [22, 129]  \n",
       "4                     [43, 150]  \n",
       "5                     [24, 131]  \n",
       "6                     [22, 129]  \n",
       "7                     [24, 131]  \n",
       "8                     [49, 156]  \n",
       "9                     [43, 150]  \n",
       "10                    [25, 132]  \n",
       "11                    [42, 149]  \n",
       "12                    [48, 155]  \n",
       "13                    [25, 132]  \n",
       "14                    [48, 155]  \n",
       "15                    [47, 154]  \n",
       "16                    [53, 160]  \n",
       "17                    [47, 154]  \n",
       "18                    [21, 128]  \n",
       "19                    [44, 151]  \n",
       "20                    [23, 130]  \n",
       "21                    [57, 164]  \n",
       "22                    [40, 147]  \n",
       "23                    [51, 158]  \n",
       "24                    [49, 156]  \n",
       "25                    [56, 163]  \n",
       "26                        [125]  \n",
       "27                    [21, 128]  \n",
       "28                    [43, 150]  \n",
       "29                    [59, 166]  \n",
       "30                    [25, 132]  \n",
       "31                    [51, 158]  \n",
       "32                    [60, 167]  \n",
       "33                    [23, 130]  \n",
       "34                    [43, 150]  \n",
       "35                    [44, 151]  \n",
       "36                    [45, 152]  \n",
       "37                    [50, 157]  \n",
       "38                    [21, 128]  \n",
       "39                    [26, 133]  \n",
       "40                    [42, 149]  \n",
       "41                    [27, 134]  \n",
       "42                    [58, 165]  \n",
       "43                    [26, 133]  \n",
       "44                    [18, 125]  \n",
       "45                    [61, 168]  \n",
       "46                    [27, 134]  \n",
       "47                    [43, 150]  \n",
       "48                    [59, 166]  \n",
       "49                    [48, 155]  \n",
       "\n",
       "[50 rows x 22 columns]"
      ]
     },
     "execution_count": 304,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "id": "839e7815-4ca6-49be-be7d-d9d81a7b9521",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Primer pair #</th>\n",
       "      <th>Forward primer Sequence (5'-&gt;3')</th>\n",
       "      <th>Forward primer Template strand</th>\n",
       "      <th>Forward primer Length</th>\n",
       "      <th>Forward primer Start</th>\n",
       "      <th>Forward primer Stop</th>\n",
       "      <th>Forward primer Tm</th>\n",
       "      <th>Forward primer GC%</th>\n",
       "      <th>Forward primer Self complementarity</th>\n",
       "      <th>Forward primer Self 3' complementarity</th>\n",
       "      <th>...</th>\n",
       "      <th>Reverse primer Length</th>\n",
       "      <th>Reverse primer Start</th>\n",
       "      <th>Reverse primer Stop</th>\n",
       "      <th>Reverse primer Tm</th>\n",
       "      <th>Reverse primer GC%</th>\n",
       "      <th>Reverse primer Self complementarity</th>\n",
       "      <th>Reverse primer Self 3' complementarity</th>\n",
       "      <th>Product length</th>\n",
       "      <th>Unnamed: 20</th>\n",
       "      <th>Variant Positions (1-based)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>CCAATCACCCGGTCTTGGTT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>10574</td>\n",
       "      <td>10593</td>\n",
       "      <td>60.25</td>\n",
       "      <td>55.00</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>10749</td>\n",
       "      <td>10729</td>\n",
       "      <td>60.20</td>\n",
       "      <td>52.38</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>176</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[84]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>TGACTTCCAATCACCCGGTC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>10568</td>\n",
       "      <td>10587</td>\n",
       "      <td>59.68</td>\n",
       "      <td>55.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>10748</td>\n",
       "      <td>10729</td>\n",
       "      <td>58.70</td>\n",
       "      <td>55.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>181</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[90]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>GACTTCCAATCACCCGGTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>10569</td>\n",
       "      <td>10588</td>\n",
       "      <td>59.39</td>\n",
       "      <td>55.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>10754</td>\n",
       "      <td>10735</td>\n",
       "      <td>61.34</td>\n",
       "      <td>60.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>186</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[89]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>4</td>\n",
       "      <td>GACTTCCAATCACCCGGTCTT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>10569</td>\n",
       "      <td>10589</td>\n",
       "      <td>60.00</td>\n",
       "      <td>52.38</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>10753</td>\n",
       "      <td>10733</td>\n",
       "      <td>59.65</td>\n",
       "      <td>57.14</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[89]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>CCAATCACCCGGTCTTGGTTA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>10574</td>\n",
       "      <td>10594</td>\n",
       "      <td>60.00</td>\n",
       "      <td>52.38</td>\n",
       "      <td>7.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>10753</td>\n",
       "      <td>10734</td>\n",
       "      <td>58.56</td>\n",
       "      <td>55.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[84]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>6</td>\n",
       "      <td>ACTTCCAATCACCCGGTCTT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>10570</td>\n",
       "      <td>10589</td>\n",
       "      <td>58.94</td>\n",
       "      <td>50.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>10749</td>\n",
       "      <td>10730</td>\n",
       "      <td>58.41</td>\n",
       "      <td>55.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[88]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>7</td>\n",
       "      <td>GTGACTTCCAATCACCCGGTC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>10567</td>\n",
       "      <td>10587</td>\n",
       "      <td>60.94</td>\n",
       "      <td>57.14</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>10752</td>\n",
       "      <td>10732</td>\n",
       "      <td>59.29</td>\n",
       "      <td>57.14</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>186</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[91]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>8</td>\n",
       "      <td>TGACTTCCAATCACCCGGTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>10568</td>\n",
       "      <td>10588</td>\n",
       "      <td>61.11</td>\n",
       "      <td>52.38</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>10754</td>\n",
       "      <td>10736</td>\n",
       "      <td>60.76</td>\n",
       "      <td>63.16</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>187</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[90]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>9</td>\n",
       "      <td>AAGTGACTTCCAATCACCCG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>10565</td>\n",
       "      <td>10584</td>\n",
       "      <td>57.81</td>\n",
       "      <td>50.00</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>10748</td>\n",
       "      <td>10728</td>\n",
       "      <td>59.15</td>\n",
       "      <td>52.38</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>184</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[93]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>10</td>\n",
       "      <td>CAATCACCCGGTCTTGGTTA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>20</td>\n",
       "      <td>10575</td>\n",
       "      <td>10594</td>\n",
       "      <td>57.52</td>\n",
       "      <td>50.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>10751</td>\n",
       "      <td>10730</td>\n",
       "      <td>59.89</td>\n",
       "      <td>54.55</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[83]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11</td>\n",
       "      <td>CACCCGGTCTTGGTTAAAATCC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>10579</td>\n",
       "      <td>10600</td>\n",
       "      <td>59.51</td>\n",
       "      <td>50.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>10754</td>\n",
       "      <td>10737</td>\n",
       "      <td>60.12</td>\n",
       "      <td>66.67</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>176</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[79]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>12</td>\n",
       "      <td>TCCAATCACCCGGTCTTGGTT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>10573</td>\n",
       "      <td>10593</td>\n",
       "      <td>61.67</td>\n",
       "      <td>52.38</td>\n",
       "      <td>7.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>10750</td>\n",
       "      <td>10730</td>\n",
       "      <td>58.86</td>\n",
       "      <td>52.38</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>178</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[85]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>13</td>\n",
       "      <td>ACTTCCAATCACCCGGTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>10570</td>\n",
       "      <td>10588</td>\n",
       "      <td>58.23</td>\n",
       "      <td>52.63</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>10748</td>\n",
       "      <td>10727</td>\n",
       "      <td>59.76</td>\n",
       "      <td>50.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>179</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[88]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>14</td>\n",
       "      <td>ACCCGGTCTTGGTTAAAATCC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>10580</td>\n",
       "      <td>10600</td>\n",
       "      <td>58.20</td>\n",
       "      <td>47.62</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>10751</td>\n",
       "      <td>10731</td>\n",
       "      <td>58.58</td>\n",
       "      <td>52.38</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>172</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[78]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>15</td>\n",
       "      <td>CAATCACCCGGTCTTGGTTAAA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>10575</td>\n",
       "      <td>10596</td>\n",
       "      <td>58.85</td>\n",
       "      <td>45.45</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>10749</td>\n",
       "      <td>10728</td>\n",
       "      <td>60.57</td>\n",
       "      <td>50.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>175</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[83]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>16</td>\n",
       "      <td>GACTTCCAATCACCCGGTC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>10569</td>\n",
       "      <td>10587</td>\n",
       "      <td>57.85</td>\n",
       "      <td>57.89</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>10750</td>\n",
       "      <td>10729</td>\n",
       "      <td>60.57</td>\n",
       "      <td>50.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>182</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[89]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>17</td>\n",
       "      <td>TAAGTGACTTCCAATCACCCG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>10564</td>\n",
       "      <td>10584</td>\n",
       "      <td>57.67</td>\n",
       "      <td>47.62</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>20</td>\n",
       "      <td>10750</td>\n",
       "      <td>10731</td>\n",
       "      <td>57.39</td>\n",
       "      <td>50.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>187</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[94]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>18</td>\n",
       "      <td>TCCAATCACCCGGTCTTGGTTA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>10573</td>\n",
       "      <td>10594</td>\n",
       "      <td>61.36</td>\n",
       "      <td>50.00</td>\n",
       "      <td>7.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>10754</td>\n",
       "      <td>10734</td>\n",
       "      <td>61.66</td>\n",
       "      <td>57.14</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>182</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[85]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>19</td>\n",
       "      <td>CAATCACCCGGTCTTGGTT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>10575</td>\n",
       "      <td>10593</td>\n",
       "      <td>57.66</td>\n",
       "      <td>52.63</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>10753</td>\n",
       "      <td>10735</td>\n",
       "      <td>58.07</td>\n",
       "      <td>57.89</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>179</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[83]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>20</td>\n",
       "      <td>TCACCCGGTCTTGGTTAAAATC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>10578</td>\n",
       "      <td>10599</td>\n",
       "      <td>58.59</td>\n",
       "      <td>45.45</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>10755</td>\n",
       "      <td>10738</td>\n",
       "      <td>60.92</td>\n",
       "      <td>72.22</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>178</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[80]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>21</td>\n",
       "      <td>CAATCACCCGGTCTTGGTTAAAA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>23</td>\n",
       "      <td>10575</td>\n",
       "      <td>10597</td>\n",
       "      <td>59.43</td>\n",
       "      <td>43.48</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>10752</td>\n",
       "      <td>10731</td>\n",
       "      <td>60.97</td>\n",
       "      <td>54.55</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>178</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[83]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>TCACCCGGTCTTGGTTAAAAT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>10578</td>\n",
       "      <td>10598</td>\n",
       "      <td>57.49</td>\n",
       "      <td>42.86</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>17</td>\n",
       "      <td>10755</td>\n",
       "      <td>10739</td>\n",
       "      <td>59.74</td>\n",
       "      <td>70.59</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>178</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[80]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>23</td>\n",
       "      <td>ATCACCCGGTCTTGGTTAAAA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>10577</td>\n",
       "      <td>10597</td>\n",
       "      <td>57.49</td>\n",
       "      <td>42.86</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>21</td>\n",
       "      <td>10747</td>\n",
       "      <td>10727</td>\n",
       "      <td>57.33</td>\n",
       "      <td>47.62</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>171</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[81]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>24</td>\n",
       "      <td>TGACTTCCAATCACCCGGTCTT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>10568</td>\n",
       "      <td>10589</td>\n",
       "      <td>61.62</td>\n",
       "      <td>50.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>17</td>\n",
       "      <td>10756</td>\n",
       "      <td>10740</td>\n",
       "      <td>60.60</td>\n",
       "      <td>76.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>189</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[90]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>25</td>\n",
       "      <td>TCACCCGGTCTTGGTTAAAATCC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>23</td>\n",
       "      <td>10578</td>\n",
       "      <td>10600</td>\n",
       "      <td>60.81</td>\n",
       "      <td>47.83</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>10755</td>\n",
       "      <td>10737</td>\n",
       "      <td>62.77</td>\n",
       "      <td>68.42</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>178</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[80]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>26</td>\n",
       "      <td>AATCACCCGGTCTTGGTTAAAA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>10576</td>\n",
       "      <td>10597</td>\n",
       "      <td>58.17</td>\n",
       "      <td>40.91</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>10750</td>\n",
       "      <td>10728</td>\n",
       "      <td>60.90</td>\n",
       "      <td>47.83</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>175</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[82]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>27</td>\n",
       "      <td>ATAAGTGACTTCCAATCACCCG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>10563</td>\n",
       "      <td>10584</td>\n",
       "      <td>58.13</td>\n",
       "      <td>45.45</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>17</td>\n",
       "      <td>10757</td>\n",
       "      <td>10741</td>\n",
       "      <td>61.01</td>\n",
       "      <td>76.47</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>195</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[95]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>28</td>\n",
       "      <td>ATCACCCGGTCTTGGTTAAAATC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>23</td>\n",
       "      <td>10577</td>\n",
       "      <td>10599</td>\n",
       "      <td>58.99</td>\n",
       "      <td>43.48</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>10753</td>\n",
       "      <td>10732</td>\n",
       "      <td>61.94</td>\n",
       "      <td>59.09</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>177</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[81]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>29</td>\n",
       "      <td>CCAATCACCCGGTCTTGGTTAAA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>23</td>\n",
       "      <td>10574</td>\n",
       "      <td>10596</td>\n",
       "      <td>61.06</td>\n",
       "      <td>47.83</td>\n",
       "      <td>7.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>10749</td>\n",
       "      <td>10727</td>\n",
       "      <td>61.09</td>\n",
       "      <td>47.83</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>176</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[84]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>30</td>\n",
       "      <td>AATCACCCGGTCTTGGTTAAA</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>10576</td>\n",
       "      <td>10596</td>\n",
       "      <td>57.49</td>\n",
       "      <td>42.86</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>10760</td>\n",
       "      <td>10745</td>\n",
       "      <td>59.33</td>\n",
       "      <td>68.75</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[82]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>31</td>\n",
       "      <td>CACCCGGTCTTGGTTAAAATC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>21</td>\n",
       "      <td>10579</td>\n",
       "      <td>10599</td>\n",
       "      <td>57.16</td>\n",
       "      <td>47.62</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>10753</td>\n",
       "      <td>10736</td>\n",
       "      <td>57.28</td>\n",
       "      <td>61.11</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>175</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[79]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>32</td>\n",
       "      <td>TTCCAATCACCCGGTCTTGGTT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>10572</td>\n",
       "      <td>10593</td>\n",
       "      <td>62.16</td>\n",
       "      <td>50.00</td>\n",
       "      <td>7.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>10751</td>\n",
       "      <td>10729</td>\n",
       "      <td>61.47</td>\n",
       "      <td>52.17</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[86]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>33</td>\n",
       "      <td>CAATCACCCGGTCTTGGTTAAAAT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>24</td>\n",
       "      <td>10575</td>\n",
       "      <td>10598</td>\n",
       "      <td>59.78</td>\n",
       "      <td>41.67</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>22</td>\n",
       "      <td>10754</td>\n",
       "      <td>10733</td>\n",
       "      <td>62.52</td>\n",
       "      <td>59.09</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[83]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>34</td>\n",
       "      <td>GTGACTTCCAATCACCCGGTCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>10567</td>\n",
       "      <td>10588</td>\n",
       "      <td>62.25</td>\n",
       "      <td>54.55</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>18</td>\n",
       "      <td>10756</td>\n",
       "      <td>10739</td>\n",
       "      <td>62.58</td>\n",
       "      <td>72.22</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[91]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>35</td>\n",
       "      <td>CTTCCAATCACCCGGTCTT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>10571</td>\n",
       "      <td>10589</td>\n",
       "      <td>57.04</td>\n",
       "      <td>52.63</td>\n",
       "      <td>5.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>17</td>\n",
       "      <td>10754</td>\n",
       "      <td>10738</td>\n",
       "      <td>58.07</td>\n",
       "      <td>70.59</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>184</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[87]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>36</td>\n",
       "      <td>AGTGACTTCCAATCACCCGGTC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>10566</td>\n",
       "      <td>10587</td>\n",
       "      <td>62.25</td>\n",
       "      <td>54.55</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>10752</td>\n",
       "      <td>10730</td>\n",
       "      <td>62.09</td>\n",
       "      <td>56.52</td>\n",
       "      <td>6.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>187</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[92]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>37</td>\n",
       "      <td>AATCACCCGGTCTTGGTTAAAAT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>23</td>\n",
       "      <td>10576</td>\n",
       "      <td>10598</td>\n",
       "      <td>58.59</td>\n",
       "      <td>39.13</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>10750</td>\n",
       "      <td>10727</td>\n",
       "      <td>61.39</td>\n",
       "      <td>45.83</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>175</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[82]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>38</td>\n",
       "      <td>AATCACCCGGTCTTGGTTAAAATC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>24</td>\n",
       "      <td>10576</td>\n",
       "      <td>10599</td>\n",
       "      <td>59.54</td>\n",
       "      <td>41.67</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>10751</td>\n",
       "      <td>10728</td>\n",
       "      <td>61.75</td>\n",
       "      <td>50.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>176</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[82]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>39</td>\n",
       "      <td>AGTGACTTCCAATCACCCG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>19</td>\n",
       "      <td>10566</td>\n",
       "      <td>10584</td>\n",
       "      <td>57.06</td>\n",
       "      <td>52.63</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>10755</td>\n",
       "      <td>10740</td>\n",
       "      <td>57.54</td>\n",
       "      <td>75.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>190</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[92]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>40</td>\n",
       "      <td>ATCACCCGGTCTTGGTTAAAAT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>22</td>\n",
       "      <td>10577</td>\n",
       "      <td>10598</td>\n",
       "      <td>57.96</td>\n",
       "      <td>40.91</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>16</td>\n",
       "      <td>10756</td>\n",
       "      <td>10741</td>\n",
       "      <td>57.54</td>\n",
       "      <td>75.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>180</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[81]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>41</td>\n",
       "      <td>TGAACTTGATTACAACAATCCTTGC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>25</td>\n",
       "      <td>10613</td>\n",
       "      <td>10637</td>\n",
       "      <td>58.55</td>\n",
       "      <td>36.00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>10811</td>\n",
       "      <td>10787</td>\n",
       "      <td>59.69</td>\n",
       "      <td>40.00</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>199</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[45]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>42</td>\n",
       "      <td>GTATAAGTGACTTCCAATCACCCG</td>\n",
       "      <td>Plus</td>\n",
       "      <td>24</td>\n",
       "      <td>10561</td>\n",
       "      <td>10584</td>\n",
       "      <td>59.19</td>\n",
       "      <td>45.83</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>25</td>\n",
       "      <td>10751</td>\n",
       "      <td>10727</td>\n",
       "      <td>62.18</td>\n",
       "      <td>48.00</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>191</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[97]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>43</td>\n",
       "      <td>AACTTGATTACAACAATCCTTGCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>24</td>\n",
       "      <td>10615</td>\n",
       "      <td>10638</td>\n",
       "      <td>57.34</td>\n",
       "      <td>33.33</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>10810</td>\n",
       "      <td>10787</td>\n",
       "      <td>58.45</td>\n",
       "      <td>41.67</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>196</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[43]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>44</td>\n",
       "      <td>GAACTTGATTACAACAATCCTTGCT</td>\n",
       "      <td>Plus</td>\n",
       "      <td>25</td>\n",
       "      <td>10614</td>\n",
       "      <td>10638</td>\n",
       "      <td>58.32</td>\n",
       "      <td>36.00</td>\n",
       "      <td>4.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>24</td>\n",
       "      <td>10811</td>\n",
       "      <td>10788</td>\n",
       "      <td>58.45</td>\n",
       "      <td>41.67</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[44]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>45</td>\n",
       "      <td>CGTATAAGTGACTTCCAATCACCC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>24</td>\n",
       "      <td>10560</td>\n",
       "      <td>10583</td>\n",
       "      <td>59.19</td>\n",
       "      <td>45.83</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>15</td>\n",
       "      <td>10759</td>\n",
       "      <td>10745</td>\n",
       "      <td>57.38</td>\n",
       "      <td>73.33</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>200</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[98]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>46</td>\n",
       "      <td>GAACTTGATTACAACAATCCTTGC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>24</td>\n",
       "      <td>10614</td>\n",
       "      <td>10637</td>\n",
       "      <td>57.07</td>\n",
       "      <td>37.50</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>10809</td>\n",
       "      <td>10787</td>\n",
       "      <td>57.17</td>\n",
       "      <td>39.13</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>196</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[44]</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>47</td>\n",
       "      <td>TTGCTATTACTATTACATTGTCCGC</td>\n",
       "      <td>Plus</td>\n",
       "      <td>25</td>\n",
       "      <td>10634</td>\n",
       "      <td>10658</td>\n",
       "      <td>57.58</td>\n",
       "      <td>36.00</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>23</td>\n",
       "      <td>10810</td>\n",
       "      <td>10788</td>\n",
       "      <td>57.13</td>\n",
       "      <td>43.48</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>177</td>\n",
       "      <td>NaN</td>\n",
       "      <td>[]</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>47 rows  22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Primer pair # Forward primer Sequence (5'->3')  \\\n",
       "0               1             CCAATCACCCGGTCTTGGTT   \n",
       "1               2             TGACTTCCAATCACCCGGTC   \n",
       "2               3             GACTTCCAATCACCCGGTCT   \n",
       "3               4            GACTTCCAATCACCCGGTCTT   \n",
       "4               5            CCAATCACCCGGTCTTGGTTA   \n",
       "5               6             ACTTCCAATCACCCGGTCTT   \n",
       "6               7            GTGACTTCCAATCACCCGGTC   \n",
       "7               8            TGACTTCCAATCACCCGGTCT   \n",
       "8               9             AAGTGACTTCCAATCACCCG   \n",
       "9              10             CAATCACCCGGTCTTGGTTA   \n",
       "10             11           CACCCGGTCTTGGTTAAAATCC   \n",
       "11             12            TCCAATCACCCGGTCTTGGTT   \n",
       "12             13              ACTTCCAATCACCCGGTCT   \n",
       "13             14            ACCCGGTCTTGGTTAAAATCC   \n",
       "14             15           CAATCACCCGGTCTTGGTTAAA   \n",
       "15             16              GACTTCCAATCACCCGGTC   \n",
       "16             17            TAAGTGACTTCCAATCACCCG   \n",
       "17             18           TCCAATCACCCGGTCTTGGTTA   \n",
       "18             19              CAATCACCCGGTCTTGGTT   \n",
       "19             20           TCACCCGGTCTTGGTTAAAATC   \n",
       "20             21          CAATCACCCGGTCTTGGTTAAAA   \n",
       "21             22            TCACCCGGTCTTGGTTAAAAT   \n",
       "22             23            ATCACCCGGTCTTGGTTAAAA   \n",
       "23             24           TGACTTCCAATCACCCGGTCTT   \n",
       "24             25          TCACCCGGTCTTGGTTAAAATCC   \n",
       "25             26           AATCACCCGGTCTTGGTTAAAA   \n",
       "26             27           ATAAGTGACTTCCAATCACCCG   \n",
       "27             28          ATCACCCGGTCTTGGTTAAAATC   \n",
       "28             29          CCAATCACCCGGTCTTGGTTAAA   \n",
       "29             30            AATCACCCGGTCTTGGTTAAA   \n",
       "30             31            CACCCGGTCTTGGTTAAAATC   \n",
       "31             32           TTCCAATCACCCGGTCTTGGTT   \n",
       "32             33         CAATCACCCGGTCTTGGTTAAAAT   \n",
       "33             34           GTGACTTCCAATCACCCGGTCT   \n",
       "34             35              CTTCCAATCACCCGGTCTT   \n",
       "35             36           AGTGACTTCCAATCACCCGGTC   \n",
       "36             37          AATCACCCGGTCTTGGTTAAAAT   \n",
       "37             38         AATCACCCGGTCTTGGTTAAAATC   \n",
       "38             39              AGTGACTTCCAATCACCCG   \n",
       "39             40           ATCACCCGGTCTTGGTTAAAAT   \n",
       "40             41        TGAACTTGATTACAACAATCCTTGC   \n",
       "41             42         GTATAAGTGACTTCCAATCACCCG   \n",
       "42             43         AACTTGATTACAACAATCCTTGCT   \n",
       "43             44        GAACTTGATTACAACAATCCTTGCT   \n",
       "44             45         CGTATAAGTGACTTCCAATCACCC   \n",
       "45             46         GAACTTGATTACAACAATCCTTGC   \n",
       "46             47        TTGCTATTACTATTACATTGTCCGC   \n",
       "\n",
       "   Forward primer Template strand  Forward primer Length  \\\n",
       "0                            Plus                     20   \n",
       "1                            Plus                     20   \n",
       "2                            Plus                     20   \n",
       "3                            Plus                     21   \n",
       "4                            Plus                     21   \n",
       "5                            Plus                     20   \n",
       "6                            Plus                     21   \n",
       "7                            Plus                     21   \n",
       "8                            Plus                     20   \n",
       "9                            Plus                     20   \n",
       "10                           Plus                     22   \n",
       "11                           Plus                     21   \n",
       "12                           Plus                     19   \n",
       "13                           Plus                     21   \n",
       "14                           Plus                     22   \n",
       "15                           Plus                     19   \n",
       "16                           Plus                     21   \n",
       "17                           Plus                     22   \n",
       "18                           Plus                     19   \n",
       "19                           Plus                     22   \n",
       "20                           Plus                     23   \n",
       "21                           Plus                     21   \n",
       "22                           Plus                     21   \n",
       "23                           Plus                     22   \n",
       "24                           Plus                     23   \n",
       "25                           Plus                     22   \n",
       "26                           Plus                     22   \n",
       "27                           Plus                     23   \n",
       "28                           Plus                     23   \n",
       "29                           Plus                     21   \n",
       "30                           Plus                     21   \n",
       "31                           Plus                     22   \n",
       "32                           Plus                     24   \n",
       "33                           Plus                     22   \n",
       "34                           Plus                     19   \n",
       "35                           Plus                     22   \n",
       "36                           Plus                     23   \n",
       "37                           Plus                     24   \n",
       "38                           Plus                     19   \n",
       "39                           Plus                     22   \n",
       "40                           Plus                     25   \n",
       "41                           Plus                     24   \n",
       "42                           Plus                     24   \n",
       "43                           Plus                     25   \n",
       "44                           Plus                     24   \n",
       "45                           Plus                     24   \n",
       "46                           Plus                     25   \n",
       "\n",
       "    Forward primer Start  Forward primer Stop  Forward primer Tm  \\\n",
       "0                  10574                10593              60.25   \n",
       "1                  10568                10587              59.68   \n",
       "2                  10569                10588              59.39   \n",
       "3                  10569                10589              60.00   \n",
       "4                  10574                10594              60.00   \n",
       "5                  10570                10589              58.94   \n",
       "6                  10567                10587              60.94   \n",
       "7                  10568                10588              61.11   \n",
       "8                  10565                10584              57.81   \n",
       "9                  10575                10594              57.52   \n",
       "10                 10579                10600              59.51   \n",
       "11                 10573                10593              61.67   \n",
       "12                 10570                10588              58.23   \n",
       "13                 10580                10600              58.20   \n",
       "14                 10575                10596              58.85   \n",
       "15                 10569                10587              57.85   \n",
       "16                 10564                10584              57.67   \n",
       "17                 10573                10594              61.36   \n",
       "18                 10575                10593              57.66   \n",
       "19                 10578                10599              58.59   \n",
       "20                 10575                10597              59.43   \n",
       "21                 10578                10598              57.49   \n",
       "22                 10577                10597              57.49   \n",
       "23                 10568                10589              61.62   \n",
       "24                 10578                10600              60.81   \n",
       "25                 10576                10597              58.17   \n",
       "26                 10563                10584              58.13   \n",
       "27                 10577                10599              58.99   \n",
       "28                 10574                10596              61.06   \n",
       "29                 10576                10596              57.49   \n",
       "30                 10579                10599              57.16   \n",
       "31                 10572                10593              62.16   \n",
       "32                 10575                10598              59.78   \n",
       "33                 10567                10588              62.25   \n",
       "34                 10571                10589              57.04   \n",
       "35                 10566                10587              62.25   \n",
       "36                 10576                10598              58.59   \n",
       "37                 10576                10599              59.54   \n",
       "38                 10566                10584              57.06   \n",
       "39                 10577                10598              57.96   \n",
       "40                 10613                10637              58.55   \n",
       "41                 10561                10584              59.19   \n",
       "42                 10615                10638              57.34   \n",
       "43                 10614                10638              58.32   \n",
       "44                 10560                10583              59.19   \n",
       "45                 10614                10637              57.07   \n",
       "46                 10634                10658              57.58   \n",
       "\n",
       "    Forward primer GC%  Forward primer Self complementarity  \\\n",
       "0                55.00                                  7.0   \n",
       "1                55.00                                  5.0   \n",
       "2                55.00                                  5.0   \n",
       "3                52.38                                  5.0   \n",
       "4                52.38                                  7.0   \n",
       "5                50.00                                  5.0   \n",
       "6                57.14                                  6.0   \n",
       "7                52.38                                  5.0   \n",
       "8                50.00                                  7.0   \n",
       "9                50.00                                  5.0   \n",
       "10               50.00                                  5.0   \n",
       "11               52.38                                  7.0   \n",
       "12               52.63                                  5.0   \n",
       "13               47.62                                  5.0   \n",
       "14               45.45                                  5.0   \n",
       "15               57.89                                  5.0   \n",
       "16               47.62                                  7.0   \n",
       "17               50.00                                  7.0   \n",
       "18               52.63                                  5.0   \n",
       "19               45.45                                  5.0   \n",
       "20               43.48                                  5.0   \n",
       "21               42.86                                  5.0   \n",
       "22               42.86                                  5.0   \n",
       "23               50.00                                  5.0   \n",
       "24               47.83                                  5.0   \n",
       "25               40.91                                  5.0   \n",
       "26               45.45                                  7.0   \n",
       "27               43.48                                  5.0   \n",
       "28               47.83                                  7.0   \n",
       "29               42.86                                  5.0   \n",
       "30               47.62                                  5.0   \n",
       "31               50.00                                  7.0   \n",
       "32               41.67                                  5.0   \n",
       "33               54.55                                  6.0   \n",
       "34               52.63                                  5.0   \n",
       "35               54.55                                  6.0   \n",
       "36               39.13                                  5.0   \n",
       "37               41.67                                  5.0   \n",
       "38               52.63                                  6.0   \n",
       "39               40.91                                  5.0   \n",
       "40               36.00                                  4.0   \n",
       "41               45.83                                  7.0   \n",
       "42               33.33                                  4.0   \n",
       "43               36.00                                  4.0   \n",
       "44               45.83                                  7.0   \n",
       "45               37.50                                  4.0   \n",
       "46               36.00                                  5.0   \n",
       "\n",
       "    Forward primer Self 3' complementarity  ... Reverse primer Length  \\\n",
       "0                                      2.0  ...                    21   \n",
       "1                                      3.0  ...                    20   \n",
       "2                                      1.0  ...                    20   \n",
       "3                                      3.0  ...                    21   \n",
       "4                                      3.0  ...                    20   \n",
       "5                                      3.0  ...                    20   \n",
       "6                                      3.0  ...                    21   \n",
       "7                                      1.0  ...                    19   \n",
       "8                                      2.0  ...                    21   \n",
       "9                                      3.0  ...                    22   \n",
       "10                                     2.0  ...                    18   \n",
       "11                                     3.0  ...                    21   \n",
       "12                                     1.0  ...                    22   \n",
       "13                                     2.0  ...                    21   \n",
       "14                                     3.0  ...                    22   \n",
       "15                                     3.0  ...                    22   \n",
       "16                                     2.0  ...                    20   \n",
       "17                                     3.0  ...                    21   \n",
       "18                                     1.0  ...                    19   \n",
       "19                                     2.0  ...                    18   \n",
       "20                                     2.0  ...                    22   \n",
       "21                                     2.0  ...                    17   \n",
       "22                                     2.0  ...                    21   \n",
       "23                                     3.0  ...                    17   \n",
       "24                                     2.0  ...                    19   \n",
       "25                                     2.0  ...                    23   \n",
       "26                                     2.0  ...                    17   \n",
       "27                                     2.0  ...                    22   \n",
       "28                                     3.0  ...                    23   \n",
       "29                                     3.0  ...                    16   \n",
       "30                                     2.0  ...                    18   \n",
       "31                                     3.0  ...                    23   \n",
       "32                                     2.0  ...                    22   \n",
       "33                                     1.0  ...                    18   \n",
       "34                                     3.0  ...                    17   \n",
       "35                                     3.0  ...                    23   \n",
       "36                                     2.0  ...                    24   \n",
       "37                                     2.0  ...                    24   \n",
       "38                                     2.0  ...                    16   \n",
       "39                                     2.0  ...                    16   \n",
       "40                                     2.0  ...                    25   \n",
       "41                                     2.0  ...                    25   \n",
       "42                                     3.0  ...                    24   \n",
       "43                                     3.0  ...                    24   \n",
       "44                                     2.0  ...                    15   \n",
       "45                                     2.0  ...                    23   \n",
       "46                                     2.0  ...                    23   \n",
       "\n",
       "   Reverse primer Start  Reverse primer Stop  Reverse primer Tm  \\\n",
       "0                 10749                10729              60.20   \n",
       "1                 10748                10729              58.70   \n",
       "2                 10754                10735              61.34   \n",
       "3                 10753                10733              59.65   \n",
       "4                 10753                10734              58.56   \n",
       "5                 10749                10730              58.41   \n",
       "6                 10752                10732              59.29   \n",
       "7                 10754                10736              60.76   \n",
       "8                 10748                10728              59.15   \n",
       "9                 10751                10730              59.89   \n",
       "10                10754                10737              60.12   \n",
       "11                10750                10730              58.86   \n",
       "12                10748                10727              59.76   \n",
       "13                10751                10731              58.58   \n",
       "14                10749                10728              60.57   \n",
       "15                10750                10729              60.57   \n",
       "16                10750                10731              57.39   \n",
       "17                10754                10734              61.66   \n",
       "18                10753                10735              58.07   \n",
       "19                10755                10738              60.92   \n",
       "20                10752                10731              60.97   \n",
       "21                10755                10739              59.74   \n",
       "22                10747                10727              57.33   \n",
       "23                10756                10740              60.60   \n",
       "24                10755                10737              62.77   \n",
       "25                10750                10728              60.90   \n",
       "26                10757                10741              61.01   \n",
       "27                10753                10732              61.94   \n",
       "28                10749                10727              61.09   \n",
       "29                10760                10745              59.33   \n",
       "30                10753                10736              57.28   \n",
       "31                10751                10729              61.47   \n",
       "32                10754                10733              62.52   \n",
       "33                10756                10739              62.58   \n",
       "34                10754                10738              58.07   \n",
       "35                10752                10730              62.09   \n",
       "36                10750                10727              61.39   \n",
       "37                10751                10728              61.75   \n",
       "38                10755                10740              57.54   \n",
       "39                10756                10741              57.54   \n",
       "40                10811                10787              59.69   \n",
       "41                10751                10727              62.18   \n",
       "42                10810                10787              58.45   \n",
       "43                10811                10788              58.45   \n",
       "44                10759                10745              57.38   \n",
       "45                10809                10787              57.17   \n",
       "46                10810                10788              57.13   \n",
       "\n",
       "    Reverse primer GC%  Reverse primer Self complementarity  \\\n",
       "0                52.38                                  6.0   \n",
       "1                55.00                                  6.0   \n",
       "2                60.00                                  6.0   \n",
       "3                57.14                                  6.0   \n",
       "4                55.00                                  6.0   \n",
       "5                55.00                                  6.0   \n",
       "6                57.14                                  6.0   \n",
       "7                63.16                                  6.0   \n",
       "8                52.38                                  6.0   \n",
       "9                54.55                                  6.0   \n",
       "10               66.67                                  6.0   \n",
       "11               52.38                                  6.0   \n",
       "12               50.00                                  6.0   \n",
       "13               52.38                                  6.0   \n",
       "14               50.00                                  6.0   \n",
       "15               50.00                                  6.0   \n",
       "16               50.00                                  6.0   \n",
       "17               57.14                                  6.0   \n",
       "18               57.89                                  6.0   \n",
       "19               72.22                                  6.0   \n",
       "20               54.55                                  6.0   \n",
       "21               70.59                                  6.0   \n",
       "22               47.62                                  4.0   \n",
       "23               76.47                                  6.0   \n",
       "24               68.42                                  6.0   \n",
       "25               47.83                                  6.0   \n",
       "26               76.47                                  6.0   \n",
       "27               59.09                                  6.0   \n",
       "28               47.83                                  6.0   \n",
       "29               68.75                                  6.0   \n",
       "30               61.11                                  6.0   \n",
       "31               52.17                                  6.0   \n",
       "32               59.09                                  6.0   \n",
       "33               72.22                                  6.0   \n",
       "34               70.59                                  6.0   \n",
       "35               56.52                                  6.0   \n",
       "36               45.83                                  6.0   \n",
       "37               50.00                                  6.0   \n",
       "38               75.00                                  6.0   \n",
       "39               75.00                                  6.0   \n",
       "40               40.00                                  2.0   \n",
       "41               48.00                                  6.0   \n",
       "42               41.67                                  2.0   \n",
       "43               41.67                                  2.0   \n",
       "44               73.33                                  6.0   \n",
       "45               39.13                                  2.0   \n",
       "46               43.48                                  2.0   \n",
       "\n",
       "    Reverse primer Self 3' complementarity  Product length  Unnamed: 20  \\\n",
       "0                                      0.0             176          NaN   \n",
       "1                                      0.0             181          NaN   \n",
       "2                                      0.0             186          NaN   \n",
       "3                                      0.0             185          NaN   \n",
       "4                                      2.0             180          NaN   \n",
       "5                                      1.0             180          NaN   \n",
       "6                                      1.0             186          NaN   \n",
       "7                                      0.0             187          NaN   \n",
       "8                                      2.0             184          NaN   \n",
       "9                                      1.0             177          NaN   \n",
       "10                                     1.0             176          NaN   \n",
       "11                                     1.0             178          NaN   \n",
       "12                                     2.0             179          NaN   \n",
       "13                                     0.0             172          NaN   \n",
       "14                                     2.0             175          NaN   \n",
       "15                                     0.0             182          NaN   \n",
       "16                                     0.0             187          NaN   \n",
       "17                                     2.0             182          NaN   \n",
       "18                                     0.0             179          NaN   \n",
       "19                                     2.0             178          NaN   \n",
       "20                                     0.0             178          NaN   \n",
       "21                                     0.0             178          NaN   \n",
       "22                                     2.0             171          NaN   \n",
       "23                                     0.0             189          NaN   \n",
       "24                                     1.0             178          NaN   \n",
       "25                                     2.0             175          NaN   \n",
       "26                                     2.0             195          NaN   \n",
       "27                                     1.0             177          NaN   \n",
       "28                                     2.0             176          NaN   \n",
       "29                                     2.0             185          NaN   \n",
       "30                                     0.0             175          NaN   \n",
       "31                                     0.0             180          NaN   \n",
       "32                                     0.0             180          NaN   \n",
       "33                                     0.0             190          NaN   \n",
       "34                                     2.0             184          NaN   \n",
       "35                                     1.0             187          NaN   \n",
       "36                                     2.0             175          NaN   \n",
       "37                                     2.0             176          NaN   \n",
       "38                                     0.0             190          NaN   \n",
       "39                                     2.0             180          NaN   \n",
       "40                                     2.0             199          NaN   \n",
       "41                                     2.0             191          NaN   \n",
       "42                                     2.0             196          NaN   \n",
       "43                                     2.0             198          NaN   \n",
       "44                                     2.0             200          NaN   \n",
       "45                                     2.0             196          NaN   \n",
       "46                                     2.0             177          NaN   \n",
       "\n",
       "    Variant Positions (1-based)  \n",
       "0                          [84]  \n",
       "1                          [90]  \n",
       "2                          [89]  \n",
       "3                          [89]  \n",
       "4                          [84]  \n",
       "5                          [88]  \n",
       "6                          [91]  \n",
       "7                          [90]  \n",
       "8                          [93]  \n",
       "9                          [83]  \n",
       "10                         [79]  \n",
       "11                         [85]  \n",
       "12                         [88]  \n",
       "13                         [78]  \n",
       "14                         [83]  \n",
       "15                         [89]  \n",
       "16                         [94]  \n",
       "17                         [85]  \n",
       "18                         [83]  \n",
       "19                         [80]  \n",
       "20                         [83]  \n",
       "21                         [80]  \n",
       "22                         [81]  \n",
       "23                         [90]  \n",
       "24                         [80]  \n",
       "25                         [82]  \n",
       "26                         [95]  \n",
       "27                         [81]  \n",
       "28                         [84]  \n",
       "29                         [82]  \n",
       "30                         [79]  \n",
       "31                         [86]  \n",
       "32                         [83]  \n",
       "33                         [91]  \n",
       "34                         [87]  \n",
       "35                         [92]  \n",
       "36                         [82]  \n",
       "37                         [82]  \n",
       "38                         [92]  \n",
       "39                         [81]  \n",
       "40                         [45]  \n",
       "41                         [97]  \n",
       "42                         [43]  \n",
       "43                         [44]  \n",
       "44                         [98]  \n",
       "45                         [44]  \n",
       "46                           []  \n",
       "\n",
       "[47 rows x 22 columns]"
      ]
     },
     "execution_count": 306,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "set2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "6bdd7d73-5f16-4f36-b11f-439da5d4a7b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from Bio.SeqUtils import MeltingTemp as mt\n",
    "\n",
    "def primer_summary(primer1, primer2, min_length=4, delta_g_threshold=-5):\n",
    "    \"\"\"\n",
    "    Evaluates potential primer dimer formation based on 3' end complementarity and G values.\n",
    "\n",
    "    Parameters:\n",
    "    primer1 (str): The sequence of the first primer.\n",
    "    primer2 (str): The sequence of the second primer.\n",
    "    min_length (int): The minimum length of complementarity at the 3' end to flag as problematic.\n",
    "    delta_g_threshold (float): The threshold for G below which dimers are considered likely.\n",
    "\n",
    "    Returns:\n",
    "    dict: A dictionary containing results for complementarity check and G evaluation.\n",
    "    \"\"\"\n",
    "    def polyx_and_melt(data, mg=4, dntp=0.4):\n",
    "        \"\"\"\n",
    "        Outputs the primer melting temperature, GC content, polyX content\n",
    "        and allows for the input of sequences, as well as Mg2+ concentrations\n",
    "        and dNTP concentrations, which may vary between master mix formulations.\n",
    "        \"\"\"\n",
    "        n_a = data.upper().count('A') #for calculating GC content\n",
    "        n_c = data.upper().count('C')\n",
    "        n_t = data.upper().count('T')\n",
    "        n_g = data.upper().count('G')\n",
    "        poly_a = data.upper().count('AAAA') #check to see if there are any polyX regions\n",
    "        poly_c = data.upper().count('CCCC')\n",
    "        poly_t = data.upper().count('TTTT')\n",
    "        poly_g = data.upper().count('GGGG')\n",
    "        poly_x = poly_a + poly_c + poly_t + poly_g\n",
    "        gc_content = (n_c + n_g) / len(data)\n",
    "        melt = round(mt.Tm_NN(data, Mg=mg, dNTPs=dntp), 2) #melt estimation\n",
    "\n",
    "        return [melt, poly_x, gc_content]\n",
    "\n",
    "    def evaluate_3_prime_complementarity(primer1, primer2, min_length=4):\n",
    "        \"\"\"\n",
    "        Check for complementarity at the 3' ends of primer1 and primer2.\n",
    "        \"\"\"\n",
    "        primer2_rc = primer2[::-1].translate(str.maketrans(\"ATGC\", \"TACG\"))\n",
    "\n",
    "        for i in range(1, len(primer1) + 1):\n",
    "            sub_seq1 = primer1[-i:]\n",
    "            sub_seq2 = primer2_rc[:i]\n",
    "\n",
    "            if len(sub_seq1) >= min_length and sub_seq1 == sub_seq2:\n",
    "                return [True]\n",
    "        return [False]\n",
    "    def clamp(primer1, primer2):\n",
    "        \"\"\"\n",
    "        Check for complementarity at the 3' ends of primer1 and primer2.\n",
    "        \"\"\" \n",
    "        terminal_c_1 = primer1[-5:].count(\"C\")\n",
    "        terminal_g_1 = primer1[-5:].count(\"G\")\n",
    "        primer2_rc = primer2[::-1].translate(str.maketrans(\"ATGC\", \"TACG\"))\n",
    "        terminal_c_2 = primer2_rc[-5:].count(\"C\")\n",
    "        terminal_g_2 = primer2_rc[-5:].count(\"G\")\n",
    "        gc_clamp_1 = False\n",
    "        gc_clamp_2 = False\n",
    "        if (terminal_c_1 + terminal_g_1) > 1 and (terminal_c_1 + terminal_g_1) < 4:\n",
    "            gc_clamp_1 = True\n",
    "        if (terminal_c_2 + terminal_g_2) > 1 and (terminal_c_2 + terminal_g_2) < 4:\n",
    "            gc_clamp_2 = True\n",
    "        return [gc_clamp_1, gc_clamp_2]\n",
    "    return [polyx_and_melt(primer1), polyx_and_melt(primer2), \n",
    "            evaluate_3_prime_complementarity(primer1, primer1), \n",
    "            evaluate_3_prime_complementarity(primer2, primer2), \n",
    "            evaluate_3_prime_complementarity(primer1, primer2),\n",
    "            clamp(primer1, primer2)]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "orbicella",
   "language": "python",
   "name": "orbicella"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
