{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "79c42bba",
   "metadata": {},
   "outputs": [],
   "source": [
    "import subprocess"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "abc9443f",
   "metadata": {},
   "outputs": [],
   "source": [
    "for sra in [\"SRR31437301\", \"SRR27608898\", \"SRR12267064\"]:\n",
    "    task = \"sra_dump\"\n",
    "    mem = \"32\"\n",
    "    cpus = \"4\"\n",
    "    prefix = sra\n",
    "    outpath = \"~/scratch/mussels/\"\n",
    "    subprocess.run([\"sbatch --mem=\" + mem + \"g --ntasks=\" + \n",
    "         cpus + \" ../\" + task + \".sh \" + \n",
    "         prefix + \" \" + outpath], shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5b36bac6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 29885432\n",
      "Submitted batch job 29885433\n",
      "Submitted batch job 29885434\n"
     ]
    }
   ],
   "source": [
    "for sra in [\"SRR31437301\", \"SRR27608898\", \"SRR12267064\"]:\n",
    "    task = \"trimgalore\"\n",
    "    mem = \"32\"\n",
    "    cpus = \"4\"\n",
    "    r1 = \"/storage/home/nsl5160/scratch/mussels/\" + sra + \"_1.fastq\"\n",
    "    r2 = \"/storage/home/nsl5160/scratch/mussels/\" + sra + \"_2.fastq\"\n",
    "    outpath = \"/storage/home/nsl5160/scratch/mussels/trimmed\"\n",
    "    outprefix = sra\n",
    "    subprocess.run([\"sbatch --mem=\" + mem + \"g --ntasks=\" + \n",
    "         cpus + \" ../\" + task + \".sh \" + \n",
    "         r1 + \" \" + r2 + \" \" + outpath + \" \" + outprefix], shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "fd9f7b82",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ERROR: This cross-compiler package contains no program /storage/group/ibb3/default/.conda/envs/ipyrad/bin/x86_64-conda_cos6-linux-gnu-cc\n",
      "ERROR: activate-gcc_linux-64.sh failed, see above for details\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR31437301_val_1.fq.gz in2=SRR31437301_val_2.fq.gz out1=../subsampled/SRR31437301_250000000_1.fq.gz out2=../subsampled/SRR31437301_250000000_2.fq.gz samplebasestarget=250000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR31437301_val_1.fq.gz, in2=SRR31437301_val_2.fq.gz, out1=../subsampled/SRR31437301_250000000_1.fq.gz, out2=../subsampled/SRR31437301_250000000_2.fq.gz, samplebasestarget=250000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t15447382 reads          \t2508479212 bases\n",
      "Output:                 \t1539460 reads (9.97%) \t250000432 bases (9.97%)\n",
      "\n",
      "Time:                         \t37.649 seconds.\n",
      "Reads Processed:      15447k \t410.30k reads/sec\n",
      "Bases Processed:       2508m \t66.63m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR31437301_val_1.fq.gz in2=SRR31437301_val_2.fq.gz out1=../subsampled/SRR31437301_500000000_1.fq.gz out2=../subsampled/SRR31437301_500000000_2.fq.gz samplebasestarget=500000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR31437301_val_1.fq.gz, in2=SRR31437301_val_2.fq.gz, out1=../subsampled/SRR31437301_500000000_1.fq.gz, out2=../subsampled/SRR31437301_500000000_2.fq.gz, samplebasestarget=500000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t15447382 reads          \t2508479212 bases\n",
      "Output:                 \t3078604 reads (19.93%) \t500000322 bases (19.93%)\n",
      "\n",
      "Time:                         \t40.145 seconds.\n",
      "Reads Processed:      15447k \t384.79k reads/sec\n",
      "Bases Processed:       2508m \t62.49m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR31437301_val_1.fq.gz in2=SRR31437301_val_2.fq.gz out1=../subsampled/SRR31437301_1000000000_1.fq.gz out2=../subsampled/SRR31437301_1000000000_2.fq.gz samplebasestarget=1000000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR31437301_val_1.fq.gz, in2=SRR31437301_val_2.fq.gz, out1=../subsampled/SRR31437301_1000000000_1.fq.gz, out2=../subsampled/SRR31437301_1000000000_2.fq.gz, samplebasestarget=1000000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t15447382 reads          \t2508479212 bases\n",
      "Output:                 \t6157284 reads (39.86%) \t1000000158 bases (39.86%)\n",
      "\n",
      "Time:                         \t47.223 seconds.\n",
      "Reads Processed:      15447k \t327.11k reads/sec\n",
      "Bases Processed:       2508m \t53.12m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR31437301_val_1.fq.gz in2=SRR31437301_val_2.fq.gz out1=../subsampled/SRR31437301_1500000000_1.fq.gz out2=../subsampled/SRR31437301_1500000000_2.fq.gz samplebasestarget=1500000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR31437301_val_1.fq.gz, in2=SRR31437301_val_2.fq.gz, out1=../subsampled/SRR31437301_1500000000_1.fq.gz, out2=../subsampled/SRR31437301_1500000000_2.fq.gz, samplebasestarget=1500000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t15447382 reads          \t2508479212 bases\n",
      "Output:                 \t9236886 reads (59.80%) \t1500000308 bases (59.80%)\n",
      "\n",
      "Time:                         \t52.136 seconds.\n",
      "Reads Processed:      15447k \t296.29k reads/sec\n",
      "Bases Processed:       2508m \t48.11m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR31437301_val_1.fq.gz in2=SRR31437301_val_2.fq.gz out1=../subsampled/SRR31437301_2500000000_1.fq.gz out2=../subsampled/SRR31437301_2500000000_2.fq.gz samplebasestarget=2500000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR31437301_val_1.fq.gz, in2=SRR31437301_val_2.fq.gz, out1=../subsampled/SRR31437301_2500000000_1.fq.gz, out2=../subsampled/SRR31437301_2500000000_2.fq.gz, samplebasestarget=2500000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t15447382 reads          \t2508479212 bases\n",
      "Output:                 \t15395266 reads (99.66%) \t2499999978 bases (99.66%)\n",
      "\n",
      "Time:                         \t63.359 seconds.\n",
      "Reads Processed:      15447k \t243.81k reads/sec\n",
      "Bases Processed:       2508m \t39.59m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR31437301_val_1.fq.gz in2=SRR31437301_val_2.fq.gz out1=../subsampled/SRR31437301_3000000000_1.fq.gz out2=../subsampled/SRR31437301_3000000000_2.fq.gz samplebasestarget=3000000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR31437301_val_1.fq.gz, in2=SRR31437301_val_2.fq.gz, out1=../subsampled/SRR31437301_3000000000_1.fq.gz, out2=../subsampled/SRR31437301_3000000000_2.fq.gz, samplebasestarget=3000000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t15447382 reads          \t2508479212 bases\n",
      "Output:                 \t15447382 reads (100.00%) \t2508479212 bases (100.00%)\n",
      "\n",
      "Time:                         \t63.869 seconds.\n",
      "Reads Processed:      15447k \t241.86k reads/sec\n",
      "Bases Processed:       2508m \t39.28m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR27608898_val_1.fq.gz in2=SRR27608898_val_2.fq.gz out1=../subsampled/SRR27608898_250000000_1.fq.gz out2=../subsampled/SRR27608898_250000000_2.fq.gz samplebasestarget=250000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR27608898_val_1.fq.gz, in2=SRR27608898_val_2.fq.gz, out1=../subsampled/SRR27608898_250000000_1.fq.gz, out2=../subsampled/SRR27608898_250000000_2.fq.gz, samplebasestarget=250000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t502533950 reads          \t71312019675 bases\n",
      "Output:                 \t1762020 reads (0.35%) \t250000077 bases (0.35%)\n",
      "\n",
      "Time:                         \t946.777 seconds.\n",
      "Reads Processed:        502m \t530.78k reads/sec\n",
      "Bases Processed:      71312m \t75.32m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR27608898_val_1.fq.gz in2=SRR27608898_val_2.fq.gz out1=../subsampled/SRR27608898_500000000_1.fq.gz out2=../subsampled/SRR27608898_500000000_2.fq.gz samplebasestarget=500000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR27608898_val_1.fq.gz, in2=SRR27608898_val_2.fq.gz, out1=../subsampled/SRR27608898_500000000_1.fq.gz, out2=../subsampled/SRR27608898_500000000_2.fq.gz, samplebasestarget=500000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input:                  \t502533950 reads          \t71312019675 bases\n",
      "Output:                 \t3524050 reads (0.70%) \t499999964 bases (0.70%)\n",
      "\n",
      "Time:                         \t956.965 seconds.\n",
      "Reads Processed:        502m \t525.13k reads/sec\n",
      "Bases Processed:      71312m \t74.52m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR27608898_val_1.fq.gz in2=SRR27608898_val_2.fq.gz out1=../subsampled/SRR27608898_1000000000_1.fq.gz out2=../subsampled/SRR27608898_1000000000_2.fq.gz samplebasestarget=1000000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR27608898_val_1.fq.gz, in2=SRR27608898_val_2.fq.gz, out1=../subsampled/SRR27608898_1000000000_1.fq.gz, out2=../subsampled/SRR27608898_1000000000_2.fq.gz, samplebasestarget=1000000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t502533950 reads          \t71312019675 bases\n",
      "Output:                 \t7047800 reads (1.40%) \t1000000220 bases (1.40%)\n",
      "\n",
      "Time:                         \t964.928 seconds.\n",
      "Reads Processed:        502m \t520.80k reads/sec\n",
      "Bases Processed:      71312m \t73.90m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR27608898_val_1.fq.gz in2=SRR27608898_val_2.fq.gz out1=../subsampled/SRR27608898_1500000000_1.fq.gz out2=../subsampled/SRR27608898_1500000000_2.fq.gz samplebasestarget=1500000000 sampleseed=1234\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Executing jgi.ReformatReads [in1=SRR27608898_val_1.fq.gz, in2=SRR27608898_val_2.fq.gz, out1=../subsampled/SRR27608898_1500000000_1.fq.gz, out2=../subsampled/SRR27608898_1500000000_2.fq.gz, samplebasestarget=1500000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t502533950 reads          \t71312019675 bases\n",
      "Output:                 \t10570756 reads (2.10%) \t1500000148 bases (2.10%)\n",
      "\n",
      "Time:                         \t1008.071 seconds.\n",
      "Reads Processed:        502m \t498.51k reads/sec\n",
      "Bases Processed:      71312m \t70.74m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR27608898_val_1.fq.gz in2=SRR27608898_val_2.fq.gz out1=../subsampled/SRR27608898_2500000000_1.fq.gz out2=../subsampled/SRR27608898_2500000000_2.fq.gz samplebasestarget=2500000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR27608898_val_1.fq.gz, in2=SRR27608898_val_2.fq.gz, out1=../subsampled/SRR27608898_2500000000_1.fq.gz, out2=../subsampled/SRR27608898_2500000000_2.fq.gz, samplebasestarget=2500000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t502533950 reads          \t71312019675 bases\n",
      "Output:                 \t17617676 reads (3.51%) \t2500000119 bases (3.51%)\n",
      "\n",
      "Time:                         \t1130.616 seconds.\n",
      "Reads Processed:        502m \t444.48k reads/sec\n",
      "Bases Processed:      71312m \t63.07m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR27608898_val_1.fq.gz in2=SRR27608898_val_2.fq.gz out1=../subsampled/SRR27608898_3000000000_1.fq.gz out2=../subsampled/SRR27608898_3000000000_2.fq.gz samplebasestarget=3000000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR27608898_val_1.fq.gz, in2=SRR27608898_val_2.fq.gz, out1=../subsampled/SRR27608898_3000000000_1.fq.gz, out2=../subsampled/SRR27608898_3000000000_2.fq.gz, samplebasestarget=3000000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t502533950 reads          \t71312019675 bases\n",
      "Output:                 \t21140926 reads (4.21%) \t3000000112 bases (4.21%)\n",
      "\n",
      "Time:                         \t1114.714 seconds.\n",
      "Reads Processed:        502m \t450.82k reads/sec\n",
      "Bases Processed:      71312m \t63.97m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR12267064_val_1.fq.gz in2=SRR12267064_val_2.fq.gz out1=../subsampled/SRR12267064_250000000_1.fq.gz out2=../subsampled/SRR12267064_250000000_2.fq.gz samplebasestarget=250000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR12267064_val_1.fq.gz, in2=SRR12267064_val_2.fq.gz, out1=../subsampled/SRR12267064_250000000_1.fq.gz, out2=../subsampled/SRR12267064_250000000_2.fq.gz, samplebasestarget=250000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t310789296 reads          \t29898867412 bases\n",
      "Output:                 \t2598802 reads (0.84%) \t250000146 bases (0.84%)\n",
      "\n",
      "Time:                         \t522.929 seconds.\n",
      "Reads Processed:        310m \t594.32k reads/sec\n",
      "Bases Processed:      29898m \t57.18m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR12267064_val_1.fq.gz in2=SRR12267064_val_2.fq.gz out1=../subsampled/SRR12267064_500000000_1.fq.gz out2=../subsampled/SRR12267064_500000000_2.fq.gz samplebasestarget=500000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR12267064_val_1.fq.gz, in2=SRR12267064_val_2.fq.gz, out1=../subsampled/SRR12267064_500000000_1.fq.gz, out2=../subsampled/SRR12267064_500000000_2.fq.gz, samplebasestarget=500000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t310789296 reads          \t29898867412 bases\n",
      "Output:                 \t5197278 reads (1.67%) \t500000111 bases (1.67%)\n",
      "\n",
      "Time:                         \t481.775 seconds.\n",
      "Reads Processed:        310m \t645.09k reads/sec\n",
      "Bases Processed:      29898m \t62.06m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR12267064_val_1.fq.gz in2=SRR12267064_val_2.fq.gz out1=../subsampled/SRR12267064_1000000000_1.fq.gz out2=../subsampled/SRR12267064_1000000000_2.fq.gz samplebasestarget=1000000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR12267064_val_1.fq.gz, in2=SRR12267064_val_2.fq.gz, out1=../subsampled/SRR12267064_1000000000_1.fq.gz, out2=../subsampled/SRR12267064_1000000000_2.fq.gz, samplebasestarget=1000000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t310789296 reads          \t29898867412 bases\n",
      "Output:                 \t10394668 reads (3.34%) \t1000000023 bases (3.34%)\n",
      "\n",
      "Time:                         \t481.459 seconds.\n",
      "Reads Processed:        310m \t645.52k reads/sec\n",
      "Bases Processed:      29898m \t62.10m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR12267064_val_1.fq.gz in2=SRR12267064_val_2.fq.gz out1=../subsampled/SRR12267064_1500000000_1.fq.gz out2=../subsampled/SRR12267064_1500000000_2.fq.gz samplebasestarget=1500000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR12267064_val_1.fq.gz, in2=SRR12267064_val_2.fq.gz, out1=../subsampled/SRR12267064_1500000000_1.fq.gz, out2=../subsampled/SRR12267064_1500000000_2.fq.gz, samplebasestarget=1500000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t310789296 reads          \t29898867412 bases\n",
      "Output:                 \t15592276 reads (5.02%) \t1500000012 bases (5.02%)\n",
      "\n",
      "Time:                         \t477.473 seconds.\n",
      "Reads Processed:        310m \t650.90k reads/sec\n",
      "Bases Processed:      29898m \t62.62m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR12267064_val_1.fq.gz in2=SRR12267064_val_2.fq.gz out1=../subsampled/SRR12267064_2500000000_1.fq.gz out2=../subsampled/SRR12267064_2500000000_2.fq.gz samplebasestarget=2500000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR12267064_val_1.fq.gz, in2=SRR12267064_val_2.fq.gz, out1=../subsampled/SRR12267064_2500000000_1.fq.gz, out2=../subsampled/SRR12267064_2500000000_2.fq.gz, samplebasestarget=2500000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t310789296 reads          \t29898867412 bases\n",
      "Output:                 \t25987286 reads (8.36%) \t2499999982 bases (8.36%)\n",
      "\n",
      "Time:                         \t532.388 seconds.\n",
      "Reads Processed:        310m \t583.76k reads/sec\n",
      "Bases Processed:      29898m \t56.16m bases/sec\n",
      "java -ea -Xms300m -cp /storage/group/ibb3/default/.conda/envs/ipyrad/opt/bbmap-38.90-0/current/ jgi.ReformatReads in1=SRR12267064_val_1.fq.gz in2=SRR12267064_val_2.fq.gz out1=../subsampled/SRR12267064_3000000000_1.fq.gz out2=../subsampled/SRR12267064_3000000000_2.fq.gz samplebasestarget=3000000000 sampleseed=1234\n",
      "Executing jgi.ReformatReads [in1=SRR12267064_val_1.fq.gz, in2=SRR12267064_val_2.fq.gz, out1=../subsampled/SRR12267064_3000000000_1.fq.gz, out2=../subsampled/SRR12267064_3000000000_2.fq.gz, samplebasestarget=3000000000, sampleseed=1234]\n",
      "\n",
      "Set INTERLEAVED to false\n",
      "Input is being processed as paired\n",
      "Input:                  \t310789296 reads          \t29898867412 bases\n",
      "Output:                 \t31184466 reads (10.03%) \t3000000080 bases (10.03%)\n",
      "\n",
      "Time:                         \t510.142 seconds.\n",
      "Reads Processed:        310m \t609.22k reads/sec\n",
      "Bases Processed:      29898m \t58.61m bases/sec\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "source ~/.bashrc\n",
    "conda activate ipyrad\n",
    "cd /storage/home/nsl5160/scratch/mussels/trimmed\n",
    "mkdir -p ../subsampled\n",
    "for SRA in SRR31437301 SRR27608898 SRR12267064; do\n",
    "    for COV in 250000000 500000000 1000000000 1500000000 2500000000 3000000000; do\n",
    "        reformat.sh \\\n",
    "            in1=${SRA}_val_1.fq.gz \\\n",
    "            in2=${SRA}_val_2.fq.gz \\\n",
    "            out1=../subsampled/${SRA}_${COV}_1.fq.gz \\\n",
    "            out2=../subsampled/${SRA}_${COV}_2.fq.gz \\\n",
    "            samplebasestarget=${COV} \\\n",
    "            sampleseed=1234\n",
    "    done\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "f783251b",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Submitted batch job 29951797\n",
      "Submitted batch job 29951798\n",
      "Submitted batch job 29951799\n",
      "Submitted batch job 29951800\n",
      "Submitted batch job 29951801\n",
      "Submitted batch job 29951802\n",
      "Submitted batch job 29951803\n",
      "Submitted batch job 29951804\n",
      "Submitted batch job 29951805\n",
      "Submitted batch job 29951806\n",
      "Submitted batch job 29951807\n",
      "Submitted batch job 29951808\n",
      "Submitted batch job 29951809\n",
      "Submitted batch job 29951810\n",
      "Submitted batch job 29951811\n",
      "Submitted batch job 29951812\n",
      "Submitted batch job 29951813\n",
      "Submitted batch job 29951814\n"
     ]
    }
   ],
   "source": [
    "for sra in [\"SRR31437301\", \"SRR27608898\", \"SRR12267064\"]:\n",
    "    for bases in [\"250000000\", \"500000000\", \"1000000000\", \"1500000000\", \"2500000000\", \"3000000000\"]:\n",
    "        task = \"mitoz_mussels\"\n",
    "        mem = \"128\"\n",
    "        cpus = \"20\"\n",
    "        r1 = '/storage/home/nsl5160/scratch/mussels/subsampled/' + sra + \"_\" + bases + \"_1.fq.gz\"\n",
    "        r2 = '/storage/home/nsl5160/scratch/mussels/subsampled/' + sra + \"_\" + bases + \"_2.fq.gz\"\n",
    "        outprefix = sra + \"_\" + bases\n",
    "        outpath = \"/storage/home/nsl5160/scratch/mussels/mitoassembly/\" + outprefix + \"/\"\n",
    "        subprocess.run([\"sbatch --mem=\" + mem + \"g --ntasks=\" + \n",
    "             cpus + \" ../\" + task + \".sh \" + \n",
    "             r1 + \" \" + r2 + \" \" + \n",
    "             outpath + \" \" + outprefix], shell=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "031beded",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/storage/home/nsl5160/scratch/mussels/mitoassembly/SRR12267064_1000000000/mt_annotation/SRR12267064_1000000000.SRR12267064_1000000000.megahit.mitogenome.fa.result/summary.txt\n",
      "#Seq_id        Length(bp)     Circularity    Closely_related_species\n",
      "k141_17214     15775          yes            Leptodea leptodon   \n",
      "/storage/home/nsl5160/scratch/mussels/mitoassembly/SRR12267064_500000000/mt_annotation/SRR12267064_500000000.SRR12267064_500000000.megahit.mitogenome.fa.result/summary.txt\n",
      "#Seq_id        Length(bp)     Circularity    Closely_related_species\n",
      "k141_6846      15775          yes            Leptodea leptodon   \n",
      "/storage/home/nsl5160/scratch/mussels/mitoassembly/SRR27608898_250000000/mt_annotation/SRR27608898_250000000.SRR27608898_250000000.megahit.mitogenome.fa.result/summary.txt\n",
      "#Seq_id        Length(bp)     Circularity    Closely_related_species\n",
      "k141_11144     17015          yes            Venustaconcha ellipsiformis\n",
      "/storage/home/nsl5160/scratch/mussels/mitoassembly/SRR27608898_500000000/mt_annotation/SRR27608898_500000000.SRR27608898_500000000.megahit.mitogenome.fa.result/summary.txt\n",
      "#Seq_id        Length(bp)     Circularity    Closely_related_species\n",
      "k141_30140     17015          yes            Venustaconcha ellipsiformis\n",
      "/storage/home/nsl5160/scratch/mussels/mitoassembly/SRR31437301_1000000000/mt_annotation/SRR31437301_1000000000.SRR31437301_1000000000.megahit.mitogenome.fa.result/summary.txt\n",
      "#Seq_id        Length(bp)     Circularity    Closely_related_species\n",
      "k141_198263    16075          yes            Lampsilis ornata    \n",
      "/storage/home/nsl5160/scratch/mussels/mitoassembly/SRR31437301_1500000000/mt_annotation/SRR31437301_1500000000.SRR31437301_1500000000.megahit.mitogenome.fa.result/summary.txt\n",
      "#Seq_id        Length(bp)     Circularity    Closely_related_species\n",
      "k141_158532    16075          yes            Lampsilis ornata    \n",
      "/storage/home/nsl5160/scratch/mussels/mitoassembly/SRR31437301_250000000/mt_annotation/SRR31437301_250000000.SRR31437301_250000000.megahit.mitogenome.fa.result/summary.txt\n",
      "#Seq_id        Length(bp)     Circularity    Closely_related_species\n",
      "k141_22496     2526           no             Lampsilis ornata    \n",
      "/storage/home/nsl5160/scratch/mussels/mitoassembly/SRR31437301_500000000/mt_annotation/SRR31437301_500000000.SRR31437301_500000000.megahit.mitogenome.fa.result/summary.txt\n",
      "#Seq_id        Length(bp)     Circularity    Closely_related_species\n",
      "k141_83623     16075          yes            Lampsilis ornata    \n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for i in $( ls ~/scratch/mussels/mitoassembly/*/mt_annotation/*/summary.txt | grep -v \"tmp\" ); do\n",
    "    echo ${i} ; grep \"Circularity\" -A1 ${i}\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f3605213",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Masters Env",
   "language": "python",
   "name": "masters"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
